{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "tE14r-GJsTUm"
      },
      "source": [
        "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mitiau/DNABERT-Z/blob/main/ZDNA-prediction.ipynb)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Тут поиск вторичных структур и подготовка генов к выравниванию"
      ],
      "metadata": {
        "id": "IPKEi9E0cjk0"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "f59Ujuujn___"
      },
      "source": [
        "# Install dependecies and define helper functions"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "apiUcTpNTnlU",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b3024c91-989b-4228-dfa5-131db5e2139f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: transformers in /usr/local/lib/python3.10/dist-packages (4.41.2)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from transformers) (3.14.0)\n",
            "Requirement already satisfied: huggingface-hub<1.0,>=0.23.0 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.23.3)\n",
            "Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from transformers) (1.25.2)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from transformers) (24.1)\n",
            "Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.10/dist-packages (from transformers) (6.0.1)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.10/dist-packages (from transformers) (2024.5.15)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from transformers) (2.31.0)\n",
            "Requirement already satisfied: tokenizers<0.20,>=0.19 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.19.1)\n",
            "Requirement already satisfied: safetensors>=0.4.1 in /usr/local/lib/python3.10/dist-packages (from transformers) (0.4.3)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.10/dist-packages (from transformers) (4.66.4)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub<1.0,>=0.23.0->transformers) (2023.6.0)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub<1.0,>=0.23.0->transformers) (4.12.2)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (3.7)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->transformers) (2024.6.2)\n",
            "Collecting biopython\n",
            "  Downloading biopython-1.83-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m20.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from biopython) (1.25.2)\n",
            "Installing collected packages: biopython\n",
            "Successfully installed biopython-1.83\n"
          ]
        }
      ],
      "source": [
        "!pip install transformers\n",
        "!pip install biopython"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "Bsyfz4BrSxMN"
      },
      "outputs": [],
      "source": [
        "import torch\n",
        "from torch import nn\n",
        "import transformers\n",
        "from transformers import BertTokenizer, BertForTokenClassification\n",
        "import numpy as np\n",
        "from Bio import SeqIO\n",
        "from io import StringIO, BytesIO\n",
        "from google.colab import drive, files\n",
        "from tqdm import tqdm\n",
        "import pickle\n",
        "import scipy\n",
        "from scipy import ndimage"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "uTiuDz7gsTUv"
      },
      "outputs": [],
      "source": [
        "def seq2kmer(seq, k):\n",
        "    kmer = [seq[x:x+k] for x in range(len(seq)+1-k)]\n",
        "    return kmer\n",
        "\n",
        "def split_seq(seq, length = 512, pad = 16):\n",
        "    res = []\n",
        "    for st in range(0, len(seq), length - pad):\n",
        "        end = min(st+512, len(seq))\n",
        "        res.append(seq[st:end])\n",
        "    return res\n",
        "\n",
        "def stitch_np_seq(np_seqs, pad = 16):\n",
        "    res = np.array([])\n",
        "    for seq in np_seqs:\n",
        "        res = res[:-pad]\n",
        "        res = np.concatenate([res,seq])\n",
        "    return res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "JWmCkPZTsTUw"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Yuz-4_vEsTUx"
      },
      "source": [
        "# Select model and parameters"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "7iqEsvPssTUx"
      },
      "outputs": [],
      "source": [
        "model = 'HG kouzine' #@param [\"HG chipseq\", \"HG kouzine\", \"MM chipseq\", \"MM kouzine\"]\n",
        "model_confidence_threshold = 0.05 #@param {type:\"number\"}\n",
        "minimum_sequence_length = 5 #@param {type:\"integer\"}"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "_arY14TzsTUy"
      },
      "outputs": [],
      "source": [
        "if model == 'HG chipseq':\n",
        "    model_id = '1VAsp8I904y_J0PUhAQqpSlCn1IqfG0FB'\n",
        "elif model == 'HG kouzine':\n",
        "    model_id = '1dAeAt5Gu2cadwDhbc7OnenUgDLHlUvkx'\n",
        "elif model == 'MM curax':\n",
        "    model_id = '1W6GEgHNoitlB-xXJbLJ_jDW4BF35W1Sd'\n",
        "elif model == 'MM kouzine':\n",
        "    model_id = '1dXpQFmheClKXIEoqcZ7kgCwx6hzVCv3H'\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "id": "9dQnWFxPsTUz",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "cf86009a-59ae-4882-d4f0-d37f8d99df69"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1dAeAt5Gu2cadwDhbc7OnenUgDLHlUvkx\n",
            "From (redirected): https://drive.google.com/uc?id=1dAeAt5Gu2cadwDhbc7OnenUgDLHlUvkx&confirm=t&uuid=2b579560-0656-47b7-935f-ab339e860dc5\n",
            "To: /content/pytorch_model.bin\n",
            "100% 354M/354M [00:04<00:00, 79.3MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=10sF8Ywktd96HqAL0CwvlZZUUGj05CGk5\n",
            "To: /content/config.json\n",
            "100% 634/634 [00:00<00:00, 2.57MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=16bT7HDv71aRwyh3gBUbKwign1mtyLD2d\n",
            "To: /content/special_tokens_map.json\n",
            "100% 112/112 [00:00<00:00, 438kB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1EE9goZ2JRSD8UTx501q71lGCk-CK3kqG\n",
            "To: /content/tokenizer_config.json\n",
            "100% 40.0/40.0 [00:00<00:00, 157kB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1gZZdtAoDnDiLQqjQfGyuwt268Pe5sXW0\n",
            "To: /content/vocab.txt\n",
            "100% 28.7k/28.7k [00:00<00:00, 51.0MB/s]\n"
          ]
        }
      ],
      "source": [
        "!gdown $model_id\n",
        "!gdown 10sF8Ywktd96HqAL0CwvlZZUUGj05CGk5\n",
        "!gdown 16bT7HDv71aRwyh3gBUbKwign1mtyLD2d\n",
        "!gdown 1EE9goZ2JRSD8UTx501q71lGCk-CK3kqG\n",
        "!gdown 1gZZdtAoDnDiLQqjQfGyuwt268Pe5sXW0\n",
        "\n",
        "\n",
        "!mkdir 6-new-12w-0\n",
        "!mv pytorch_model.bin 6-new-12w-0/\n",
        "!mv config.json 6-new-12w-0/\n",
        "!mv special_tokens_map.json 6-new-12w-0/\n",
        "!mv tokenizer_config.json 6-new-12w-0/\n",
        "!mv vocab.txt 6-new-12w-0/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "id": "25r7It-vsTUz",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "58e0158d-88b3-4764-8321-4ac6badce770"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "BertForTokenClassification(\n",
              "  (bert): BertModel(\n",
              "    (embeddings): BertEmbeddings(\n",
              "      (word_embeddings): Embedding(4101, 768, padding_idx=0)\n",
              "      (position_embeddings): Embedding(512, 768)\n",
              "      (token_type_embeddings): Embedding(2, 768)\n",
              "      (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
              "      (dropout): Dropout(p=0.1, inplace=False)\n",
              "    )\n",
              "    (encoder): BertEncoder(\n",
              "      (layer): ModuleList(\n",
              "        (0-11): 12 x BertLayer(\n",
              "          (attention): BertAttention(\n",
              "            (self): BertSdpaSelfAttention(\n",
              "              (query): Linear(in_features=768, out_features=768, bias=True)\n",
              "              (key): Linear(in_features=768, out_features=768, bias=True)\n",
              "              (value): Linear(in_features=768, out_features=768, bias=True)\n",
              "              (dropout): Dropout(p=0.1, inplace=False)\n",
              "            )\n",
              "            (output): BertSelfOutput(\n",
              "              (dense): Linear(in_features=768, out_features=768, bias=True)\n",
              "              (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
              "              (dropout): Dropout(p=0.1, inplace=False)\n",
              "            )\n",
              "          )\n",
              "          (intermediate): BertIntermediate(\n",
              "            (dense): Linear(in_features=768, out_features=3072, bias=True)\n",
              "            (intermediate_act_fn): GELUActivation()\n",
              "          )\n",
              "          (output): BertOutput(\n",
              "            (dense): Linear(in_features=3072, out_features=768, bias=True)\n",
              "            (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
              "            (dropout): Dropout(p=0.1, inplace=False)\n",
              "          )\n",
              "        )\n",
              "      )\n",
              "    )\n",
              "  )\n",
              "  (dropout): Dropout(p=0.1, inplace=False)\n",
              "  (classifier): Linear(in_features=768, out_features=2, bias=True)\n",
              ")"
            ]
          },
          "metadata": {},
          "execution_count": 7
        }
      ],
      "source": [
        "tokenizer = BertTokenizer.from_pretrained('6-new-12w-0/')\n",
        "model = BertForTokenClassification.from_pretrained('6-new-12w-0/')\n",
        "model.cuda()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "NzLQYqY6sTU0"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TAp3NQyaupxE"
      },
      "source": [
        "# Upload fasta files for prediction"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "id": "urrTMPfMUrbP",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 412
        },
        "outputId": "be3fa6d0-a1d7-404e-a4bb-82f62b7d594b"
      },
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.HTML object>"
            ],
            "text/html": [
              "\n",
              "     <input type=\"file\" id=\"files-c290da66-5802-43ab-be29-944a3aba499f\" name=\"files[]\" multiple disabled\n",
              "        style=\"border:none\" />\n",
              "     <output id=\"result-c290da66-5802-43ab-be29-944a3aba499f\">\n",
              "      Upload widget is only available when the cell has been executed in the\n",
              "      current browser session. Please rerun this cell to enable.\n",
              "      </output>\n",
              "      <script>// Copyright 2017 Google LLC\n",
              "//\n",
              "// Licensed under the Apache License, Version 2.0 (the \"License\");\n",
              "// you may not use this file except in compliance with the License.\n",
              "// You may obtain a copy of the License at\n",
              "//\n",
              "//      http://www.apache.org/licenses/LICENSE-2.0\n",
              "//\n",
              "// Unless required by applicable law or agreed to in writing, software\n",
              "// distributed under the License is distributed on an \"AS IS\" BASIS,\n",
              "// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
              "// See the License for the specific language governing permissions and\n",
              "// limitations under the License.\n",
              "\n",
              "/**\n",
              " * @fileoverview Helpers for google.colab Python module.\n",
              " */\n",
              "(function(scope) {\n",
              "function span(text, styleAttributes = {}) {\n",
              "  const element = document.createElement('span');\n",
              "  element.textContent = text;\n",
              "  for (const key of Object.keys(styleAttributes)) {\n",
              "    element.style[key] = styleAttributes[key];\n",
              "  }\n",
              "  return element;\n",
              "}\n",
              "\n",
              "// Max number of bytes which will be uploaded at a time.\n",
              "const MAX_PAYLOAD_SIZE = 100 * 1024;\n",
              "\n",
              "function _uploadFiles(inputId, outputId) {\n",
              "  const steps = uploadFilesStep(inputId, outputId);\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  // Cache steps on the outputElement to make it available for the next call\n",
              "  // to uploadFilesContinue from Python.\n",
              "  outputElement.steps = steps;\n",
              "\n",
              "  return _uploadFilesContinue(outputId);\n",
              "}\n",
              "\n",
              "// This is roughly an async generator (not supported in the browser yet),\n",
              "// where there are multiple asynchronous steps and the Python side is going\n",
              "// to poll for completion of each step.\n",
              "// This uses a Promise to block the python side on completion of each step,\n",
              "// then passes the result of the previous step as the input to the next step.\n",
              "function _uploadFilesContinue(outputId) {\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  const steps = outputElement.steps;\n",
              "\n",
              "  const next = steps.next(outputElement.lastPromiseValue);\n",
              "  return Promise.resolve(next.value.promise).then((value) => {\n",
              "    // Cache the last promise value to make it available to the next\n",
              "    // step of the generator.\n",
              "    outputElement.lastPromiseValue = value;\n",
              "    return next.value.response;\n",
              "  });\n",
              "}\n",
              "\n",
              "/**\n",
              " * Generator function which is called between each async step of the upload\n",
              " * process.\n",
              " * @param {string} inputId Element ID of the input file picker element.\n",
              " * @param {string} outputId Element ID of the output display.\n",
              " * @return {!Iterable<!Object>} Iterable of next steps.\n",
              " */\n",
              "function* uploadFilesStep(inputId, outputId) {\n",
              "  const inputElement = document.getElementById(inputId);\n",
              "  inputElement.disabled = false;\n",
              "\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  outputElement.innerHTML = '';\n",
              "\n",
              "  const pickedPromise = new Promise((resolve) => {\n",
              "    inputElement.addEventListener('change', (e) => {\n",
              "      resolve(e.target.files);\n",
              "    });\n",
              "  });\n",
              "\n",
              "  const cancel = document.createElement('button');\n",
              "  inputElement.parentElement.appendChild(cancel);\n",
              "  cancel.textContent = 'Cancel upload';\n",
              "  const cancelPromise = new Promise((resolve) => {\n",
              "    cancel.onclick = () => {\n",
              "      resolve(null);\n",
              "    };\n",
              "  });\n",
              "\n",
              "  // Wait for the user to pick the files.\n",
              "  const files = yield {\n",
              "    promise: Promise.race([pickedPromise, cancelPromise]),\n",
              "    response: {\n",
              "      action: 'starting',\n",
              "    }\n",
              "  };\n",
              "\n",
              "  cancel.remove();\n",
              "\n",
              "  // Disable the input element since further picks are not allowed.\n",
              "  inputElement.disabled = true;\n",
              "\n",
              "  if (!files) {\n",
              "    return {\n",
              "      response: {\n",
              "        action: 'complete',\n",
              "      }\n",
              "    };\n",
              "  }\n",
              "\n",
              "  for (const file of files) {\n",
              "    const li = document.createElement('li');\n",
              "    li.append(span(file.name, {fontWeight: 'bold'}));\n",
              "    li.append(span(\n",
              "        `(${file.type || 'n/a'}) - ${file.size} bytes, ` +\n",
              "        `last modified: ${\n",
              "            file.lastModifiedDate ? file.lastModifiedDate.toLocaleDateString() :\n",
              "                                    'n/a'} - `));\n",
              "    const percent = span('0% done');\n",
              "    li.appendChild(percent);\n",
              "\n",
              "    outputElement.appendChild(li);\n",
              "\n",
              "    const fileDataPromise = new Promise((resolve) => {\n",
              "      const reader = new FileReader();\n",
              "      reader.onload = (e) => {\n",
              "        resolve(e.target.result);\n",
              "      };\n",
              "      reader.readAsArrayBuffer(file);\n",
              "    });\n",
              "    // Wait for the data to be ready.\n",
              "    let fileData = yield {\n",
              "      promise: fileDataPromise,\n",
              "      response: {\n",
              "        action: 'continue',\n",
              "      }\n",
              "    };\n",
              "\n",
              "    // Use a chunked sending to avoid message size limits. See b/62115660.\n",
              "    let position = 0;\n",
              "    do {\n",
              "      const length = Math.min(fileData.byteLength - position, MAX_PAYLOAD_SIZE);\n",
              "      const chunk = new Uint8Array(fileData, position, length);\n",
              "      position += length;\n",
              "\n",
              "      const base64 = btoa(String.fromCharCode.apply(null, chunk));\n",
              "      yield {\n",
              "        response: {\n",
              "          action: 'append',\n",
              "          file: file.name,\n",
              "          data: base64,\n",
              "        },\n",
              "      };\n",
              "\n",
              "      let percentDone = fileData.byteLength === 0 ?\n",
              "          100 :\n",
              "          Math.round((position / fileData.byteLength) * 100);\n",
              "      percent.textContent = `${percentDone}% done`;\n",
              "\n",
              "    } while (position < fileData.byteLength);\n",
              "  }\n",
              "\n",
              "  // All done.\n",
              "  yield {\n",
              "    response: {\n",
              "      action: 'complete',\n",
              "    }\n",
              "  };\n",
              "}\n",
              "\n",
              "scope.google = scope.google || {};\n",
              "scope.google.colab = scope.google.colab || {};\n",
              "scope.google.colab._files = {\n",
              "  _uploadFiles,\n",
              "  _uploadFilesContinue,\n",
              "};\n",
              "})(self);\n",
              "</script> "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "error",
          "ename": "KeyboardInterrupt",
          "evalue": "",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-8-633487f4966c>\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0muploaded\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfiles\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mfn\u001b[0m \u001b[0;32min\u001b[0m \u001b[0muploaded\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m   print('User uploaded file \"{name}\" with length {length} bytes'.format(\n\u001b[1;32m      4\u001b[0m       name=fn, length=len(uploaded[fn])))\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/google/colab/files.py\u001b[0m in \u001b[0;36mupload\u001b[0;34m()\u001b[0m\n\u001b[1;32m     67\u001b[0m   \"\"\"\n\u001b[1;32m     68\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 69\u001b[0;31m   \u001b[0muploaded_files\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_upload_files\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmultiple\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     70\u001b[0m   \u001b[0;31m# Mapping from original filename to filename as saved locally.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     71\u001b[0m   \u001b[0mlocal_filenames\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/google/colab/files.py\u001b[0m in \u001b[0;36m_upload_files\u001b[0;34m(multiple)\u001b[0m\n\u001b[1;32m    162\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    163\u001b[0m   \u001b[0;32mwhile\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'action'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;34m'complete'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 164\u001b[0;31m     result = _output.eval_js(\n\u001b[0m\u001b[1;32m    165\u001b[0m         'google.colab._files._uploadFilesContinue(\"{output_id}\")'.format(\n\u001b[1;32m    166\u001b[0m             \u001b[0moutput_id\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0moutput_id\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/google/colab/output/_js.py\u001b[0m in \u001b[0;36meval_js\u001b[0;34m(script, ignore_result, timeout_sec)\u001b[0m\n\u001b[1;32m     38\u001b[0m   \u001b[0;32mif\u001b[0m \u001b[0mignore_result\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     39\u001b[0m     \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 40\u001b[0;31m   \u001b[0;32mreturn\u001b[0m \u001b[0m_message\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_reply_from_input\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout_sec\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     41\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     42\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/google/colab/_message.py\u001b[0m in \u001b[0;36mread_reply_from_input\u001b[0;34m(message_id, timeout_sec)\u001b[0m\n\u001b[1;32m     94\u001b[0m     \u001b[0mreply\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_read_next_input_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     95\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mreply\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0m_NOT_READY\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreply\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 96\u001b[0;31m       \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msleep\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0.025\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     97\u001b[0m       \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     98\u001b[0m     if (\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
          ]
        }
      ],
      "source": [
        "uploaded = files.upload()\n",
        "for fn in uploaded.keys():\n",
        "  print('User uploaded file \"{name}\" with length {length} bytes'.format(\n",
        "      name=fn, length=len(uploaded[fn])))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "4SB_6nhysTU0"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "KW5PzHQasTU1"
      },
      "source": [
        "# Predict and save raw outputs"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "_1qjq3i-VCEz"
      },
      "outputs": [],
      "source": [
        "out = []\n",
        "for key in uploaded.keys():\n",
        "    print(key)\n",
        "    out.append(key)\n",
        "    result_dict = {}\n",
        "    for seq_record in SeqIO.parse(StringIO(BytesIO(uploaded[key]).read().decode('UTF-8')), 'fasta'):\n",
        "        kmer_seq = seq2kmer(str(seq_record.seq).upper(), 6)\n",
        "        seq_pieces = split_seq(kmer_seq)\n",
        "        print(seq_record.name)\n",
        "        out.append(seq_record.name)\n",
        "        with torch.no_grad():\n",
        "            preds = []\n",
        "            for seq_piece in tqdm(seq_pieces):\n",
        "                input_ids = torch.LongTensor(tokenizer.encode(' '.join(seq_piece), add_special_tokens=False))\n",
        "                outputs = torch.softmax(model(input_ids.cuda().unsqueeze(0))[-1],axis = -1)[0,:,1]\n",
        "                preds.append(outputs.cpu().numpy())\n",
        "        result_dict[seq_record.name] = stitch_np_seq(preds)\n",
        "\n",
        "\n",
        "\n",
        "        labeled, max_label = scipy.ndimage.label(result_dict[seq_record.name]>model_confidence_threshold)\n",
        "        print('  start     end')\n",
        "        out.append('  start     end')\n",
        "        for label in range(1, max_label+1):\n",
        "            candidate = np.where(labeled == label)[0]\n",
        "            candidate_length = candidate.shape[0]\n",
        "            if candidate_length>minimum_sequence_length:\n",
        "                print('{:8}'.format(candidate[0]), '{:8}'.format(candidate[-1]))\n",
        "                out.append('{:8}'.format(candidate[0]) + '{:8}'.format(candidate[-1]))\n",
        "\n",
        "    with open(key + '.preds.pkl',\"wb\") as fh:\n",
        "      pickle.dump(result_dict, fh)\n",
        "    print()\n",
        "\n",
        "with open('text_predictions.txt',\"w\") as fh:\n",
        "    for item in out:\n",
        "        fh.write(\"%s\\n\" % item)\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "rKA52S_RsTU1"
      },
      "source": [
        "# Download text file with predictions"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "n4PvzU0EsTU1"
      },
      "outputs": [],
      "source": [
        "files.download('text_predictions.txt')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "A2ExoqPUsTU2"
      },
      "source": [
        "# Download raw prediction files in numpy format"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "4by6H7tRdcQO",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 17
        },
        "outputId": "99b4014c-3c0e-4880-8412-d0aeb3b70925"
      },
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_82cde78a-2748-45ee-af98-d800fdc0d1ca\", \"GCF_000189635.1_JCVI-TTA1-2.2_genomic.fna.preds.pkl\", 824130306)"
            ]
          },
          "metadata": {}
        }
      ],
      "source": [
        "for key in uploaded.keys():\n",
        "    files.download(key + '.preds.pkl')"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "uploaded.keys()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 141
        },
        "id": "Cx6EqXy5Rov2",
        "outputId": "b2359ed8-fbc0-466c-fcd8-868aabd9cc84"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "error",
          "ename": "NameError",
          "evalue": "name 'uploaded' is not defined",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-10-4d0baf6b608f>\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0muploaded\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
            "\u001b[0;31mNameError\u001b[0m: name 'uploaded' is not defined"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Скачиваем несколько необходимых файлов"
      ],
      "metadata": {
        "id": "UZqa6LeYczNS"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!gdown 1tvJDi-DYx3NEescA7AB1tfwTMMMIk90Z\n",
        "!gdown 1bhf5HhzDmFzS01hfwFPOHhzjl2KX20zP\n",
        "!gdown 1P-jBUfISSKbnaixTY6ehic4r-w-PCO5Z"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "txS3n9T85DYH",
        "outputId": "e5c05e92-666e-416b-9d5d-dbe4ee0784a2"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1tvJDi-DYx3NEescA7AB1tfwTMMMIk90Z\n",
            "From (redirected): https://drive.google.com/uc?id=1tvJDi-DYx3NEescA7AB1tfwTMMMIk90Z&confirm=t&uuid=86eb98d4-7bde-409d-924f-a62859c1f4e0\n",
            "To: /content/GCA_903995115.1_Adeanei_nanopore_chromosomes_genomic.fna.Z-SCORE\n",
            "100% 1.90G/1.90G [00:32<00:00, 57.9MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1bhf5HhzDmFzS01hfwFPOHhzjl2KX20zP\n",
            "To: /content/text_predictions.txt\n",
            "100% 252k/252k [00:00<00:00, 122MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1P-jBUfISSKbnaixTY6ehic4r-w-PCO5Z\n",
            "To: /content/GCA_903995115.1_Adeanei_nanopore_chromosomes_genomic.fna\n",
            "100% 21.2M/21.2M [00:00<00:00, 23.0MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Далее поиск квадруплексов"
      ],
      "metadata": {
        "id": "7fe6Z_25c7-m"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import re\n",
        "from Bio import SeqIO\n",
        "\n",
        "mas = []\n",
        "\n",
        "pattern=\"(?:G{3,}[ATGC]{1,7}){3,}G{3,}\"\n",
        "pattern_minus = \"(?:C{3,}[ATGC]{1,7}){3,}C{3,}\"\n",
        "for record in SeqIO.parse(\"GCA_905183005.1_Paramecium_pentaurelia_V1_genomic.fna\",'fasta'):\n",
        "  for m in re.finditer(pattern, str(record.seq),re.IGNORECASE):\n",
        "    mas.append([record.id, m.start(),m.end(),m.group(0)])\n",
        "  for m in re.finditer(pattern_minus, str(record.seq),re.IGNORECASE):\n",
        "    mas.append([record.id, m.start(),m.end(),m.group(0)])\n",
        "\n",
        "with open(\"pqs.bed\", \"w\") as f:\n",
        "  for i in mas:\n",
        "    f.write(f\"{i[0]}\\t{i[1]}\\t{i[2]}\\n\")"
      ],
      "metadata": {
        "id": "jnge_mVrLxG4"
      },
      "execution_count": 24,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Скачиваем bedtools"
      ],
      "metadata": {
        "id": "AFkxR8pudMaJ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
        "!tar -zxvf bedtools-2.30.0.tar.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6OMlKOe3GYnS",
        "outputId": "04eeddd8-660c-46df-a34f-5c11cb5cc3db"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2024-06-16 12:41:16--  http://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
            "Resolving github.com (github.com)... 20.205.243.166\n",
            "Connecting to github.com (github.com)|20.205.243.166|:80... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz [following]\n",
            "--2024-06-16 12:41:16--  https://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
            "Connecting to github.com (github.com)|20.205.243.166|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/15059334/aecc9080-5d79-11eb-847e-0ff72e8f556a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240616%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240616T124116Z&X-Amz-Expires=300&X-Amz-Signature=516c94cb26db62daf31e90279a135f5b3e57c5a99c87972e4b4930d8b9ca25ca&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=15059334&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.30.0.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2024-06-16 12:41:16--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/15059334/aecc9080-5d79-11eb-847e-0ff72e8f556a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240616%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240616T124116Z&X-Amz-Expires=300&X-Amz-Signature=516c94cb26db62daf31e90279a135f5b3e57c5a99c87972e4b4930d8b9ca25ca&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=15059334&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.30.0.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.111.133, 185.199.108.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.111.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 20648577 (20M) [application/octet-stream]\n",
            "Saving to: ‘bedtools-2.30.0.tar.gz’\n",
            "\n",
            "bedtools-2.30.0.tar 100%[===================>]  19.69M  9.41MB/s    in 2.1s    \n",
            "\n",
            "2024-06-16 12:41:19 (9.41 MB/s) - ‘bedtools-2.30.0.tar.gz’ saved [20648577/20648577]\n",
            "\n",
            "bedtools2/\n",
            "bedtools2/obj/\n",
            "bedtools2/tutorial/\n",
            "bedtools2/LICENSE\n",
            "bedtools2/test/\n",
            "bedtools2/bin/\n",
            "bedtools2/Makefile\n",
            "bedtools2/docs/\n",
            "bedtools2/README.md\n",
            "bedtools2/genomes/\n",
            "bedtools2/.gitignore\n",
            "bedtools2/scripts/\n",
            "bedtools2/.github/\n",
            "bedtools2/data/\n",
            "bedtools2/src/\n",
            "bedtools2/src/spacingFile/\n",
            "bedtools2/src/sortBed/\n",
            "bedtools2/src/split/\n",
            "bedtools2/src/bedToBam/\n",
            "bedtools2/src/complementFile/\n",
            "bedtools2/src/expand/\n",
            "bedtools2/src/fisher/\n",
            "bedtools2/src/linksBed/\n",
            "bedtools2/src/pairToPair/\n",
            "bedtools2/src/bedtools.cpp\n",
            "bedtools2/src/mapFile/\n",
            "bedtools2/src/slopBed/\n",
            "bedtools2/src/annotateBed/\n",
            "bedtools2/src/randomBed/\n",
            "bedtools2/src/getOverlap/\n",
            "bedtools2/src/reldist/\n",
            "bedtools2/src/nucBed/\n",
            "bedtools2/src/bedToIgv/\n",
            "bedtools2/src/regressTest/\n",
            "bedtools2/src/maskFastaFromBed/\n",
            "bedtools2/src/genomeCoverageBed/\n",
            "bedtools2/src/utils/\n",
            "bedtools2/src/windowBed/\n",
            "bedtools2/src/jaccard/\n",
            "bedtools2/src/groupBy/\n",
            "bedtools2/src/bamToFastq/\n",
            "bedtools2/src/htsutil.cpp\n",
            "bedtools2/src/fastaFromBed/\n",
            "bedtools2/src/nekSandbox1/\n",
            "bedtools2/src/unionBedGraphs/\n",
            "bedtools2/src/bed12ToBed6/\n",
            "bedtools2/src/subtractFile/\n",
            "bedtools2/src/bedpeToBam/\n",
            "bedtools2/src/windowMaker/\n",
            "bedtools2/src/coverageFile/\n",
            "bedtools2/src/summaryFile/\n",
            "bedtools2/src/multiIntersectBed/\n",
            "bedtools2/src/intersectFile/\n",
            "bedtools2/src/closestFile/\n",
            "bedtools2/src/shuffleBed/\n",
            "bedtools2/src/flankBed/\n",
            "bedtools2/src/tagBam/\n",
            "bedtools2/src/multiBamCov/\n",
            "bedtools2/src/sampleFile/\n",
            "bedtools2/src/clusterBed/\n",
            "bedtools2/src/shiftBed/\n",
            "bedtools2/src/mergeFile/\n",
            "bedtools2/src/bamToBed/\n",
            "bedtools2/src/pairToBed/\n",
            "bedtools2/src/pairToBed/pairToBed.h\n",
            "bedtools2/src/pairToBed/pairToBedMain.cpp\n",
            "bedtools2/src/pairToBed/Makefile.frag\n",
            "bedtools2/src/pairToBed/pairToBed.cpp\n",
            "bedtools2/src/bamToBed/bamToBed.cpp\n",
            "bedtools2/src/bamToBed/Makefile.frag\n",
            "bedtools2/src/mergeFile/mergeFile.cpp\n",
            "bedtools2/src/mergeFile/Makefile.frag\n",
            "bedtools2/src/mergeFile/mergeHelp.cpp\n",
            "bedtools2/src/mergeFile/mergeFile.h\n",
            "bedtools2/src/shiftBed/shiftBed.cpp\n",
            "bedtools2/src/shiftBed/shiftBedMain.cpp\n",
            "bedtools2/src/shiftBed/Makefile.frag\n",
            "bedtools2/src/shiftBed/shiftBed.h\n",
            "bedtools2/src/clusterBed/clusterBed.h\n",
            "bedtools2/src/clusterBed/clusterBed.cpp\n",
            "bedtools2/src/clusterBed/clusterMain.cpp\n",
            "bedtools2/src/clusterBed/Makefile.frag\n",
            "bedtools2/src/sampleFile/sampleFile.cpp\n",
            "bedtools2/src/sampleFile/sampleFile.h\n",
            "bedtools2/src/sampleFile/Makefile.frag\n",
            "bedtools2/src/sampleFile/sampleHelp.cpp\n",
            "bedtools2/src/multiBamCov/multiBamCov.cpp\n",
            "bedtools2/src/multiBamCov/Makefile.frag\n",
            "bedtools2/src/multiBamCov/multiBamCov.h\n",
            "bedtools2/src/multiBamCov/multiBamCovMain.cpp\n",
            "bedtools2/src/tagBam/tagBam.cpp\n",
            "bedtools2/src/tagBam/tagBamMain.cpp\n",
            "bedtools2/src/tagBam/tagBam.h\n",
            "bedtools2/src/tagBam/Makefile.frag\n",
            "bedtools2/src/flankBed/flankBed.cpp\n",
            "bedtools2/src/flankBed/flankBedMain.cpp\n",
            "bedtools2/src/flankBed/test.fa.fai\n",
            "bedtools2/src/flankBed/test.gff\n",
            "bedtools2/src/flankBed/Makefile.frag\n",
            "bedtools2/src/flankBed/test.fa\n",
            "bedtools2/src/flankBed/flankBed.h\n",
            "bedtools2/src/shuffleBed/shuffleBed.h\n",
            "bedtools2/src/shuffleBed/shuffleBed.cpp\n",
            "bedtools2/src/shuffleBed/Makefile.frag\n",
            "bedtools2/src/shuffleBed/shuffleBedMain.cpp\n",
            "bedtools2/src/closestFile/closestHelp.cpp\n",
            "bedtools2/src/closestFile/closestFile.h\n",
            "bedtools2/src/closestFile/closestFile.cpp\n",
            "bedtools2/src/closestFile/Makefile.frag\n",
            "bedtools2/src/intersectFile/intersectHelp.cpp\n",
            "bedtools2/src/intersectFile/Makefile.frag\n",
            "bedtools2/src/intersectFile/intersectFile.cpp\n",
            "bedtools2/src/intersectFile/intersectFile.h\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBed.cpp\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBed.h\n",
            "bedtools2/src/multiIntersectBed/Makefile.frag\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "bedtools2/src/summaryFile/summaryHelp.cpp\n",
            "bedtools2/src/summaryFile/Makefile.frag\n",
            "bedtools2/src/summaryFile/summaryFile.h\n",
            "bedtools2/src/summaryFile/summaryFile.cpp\n",
            "bedtools2/src/coverageFile/coverageFile.cpp\n",
            "bedtools2/src/coverageFile/coverageFile.h\n",
            "bedtools2/src/coverageFile/Makefile.frag\n",
            "bedtools2/src/coverageFile/coverageHelp.cpp\n",
            "bedtools2/src/windowMaker/windowMakerMain.cpp\n",
            "bedtools2/src/windowMaker/windowMaker.h\n",
            "bedtools2/src/windowMaker/windowMaker.cpp\n",
            "bedtools2/src/windowMaker/Makefile.frag\n",
            "bedtools2/src/bedpeToBam/notes.txt\n",
            "bedtools2/src/bedpeToBam/bedpeToBam.cpp\n",
            "bedtools2/src/bedpeToBam/Makefile.frag\n",
            "bedtools2/src/subtractFile/subtractHelp.cpp\n",
            "bedtools2/src/subtractFile/subtractFile.h\n",
            "bedtools2/src/subtractFile/Makefile.frag\n",
            "bedtools2/src/subtractFile/subtractFile.cpp\n",
            "bedtools2/src/bed12ToBed6/bed12ToBed6.cpp\n",
            "bedtools2/src/bed12ToBed6/Makefile.frag\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphs.cpp\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "bedtools2/src/unionBedGraphs/Makefile.frag\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphs.h\n",
            "bedtools2/src/nekSandbox1/FormatGuess.h\n",
            "bedtools2/src/nekSandbox1/Makefile\n",
            "bedtools2/src/nekSandbox1/nekSandboxMain.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBedMain.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBed.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBed.h\n",
            "bedtools2/src/fastaFromBed/Makefile.frag\n",
            "bedtools2/src/bamToFastq/bamToFastq.h\n",
            "bedtools2/src/bamToFastq/bamToFastq.cpp\n",
            "bedtools2/src/bamToFastq/bamToFastqMain.cpp\n",
            "bedtools2/src/bamToFastq/Makefile.frag\n",
            "bedtools2/src/groupBy/groupBy.cpp\n",
            "bedtools2/src/groupBy/groupBy.h\n",
            "bedtools2/src/groupBy/Makefile.frag\n",
            "bedtools2/src/groupBy/groupByHelp.cpp\n",
            "bedtools2/src/jaccard/jaccard.cpp\n",
            "bedtools2/src/jaccard/Makefile.frag\n",
            "bedtools2/src/jaccard/jaccardHelp.cpp\n",
            "bedtools2/src/jaccard/jaccard.h\n",
            "bedtools2/src/windowBed/windowBed.cpp\n",
            "bedtools2/src/windowBed/windowBed.h\n",
            "bedtools2/src/windowBed/windowMain.cpp\n",
            "bedtools2/src/windowBed/Makefile.frag\n",
            "bedtools2/src/utils/tabFile/\n",
            "bedtools2/src/utils/htslib/\n",
            "bedtools2/src/utils/NewChromsweep/\n",
            "bedtools2/src/utils/RecordOutputMgr/\n",
            "bedtools2/src/utils/bedFile/\n",
            "bedtools2/src/utils/Contexts/\n",
            "bedtools2/src/utils/BamTools/\n",
            "bedtools2/src/utils/BlockedIntervals/\n",
            "bedtools2/src/utils/general/\n",
            "bedtools2/src/utils/driver/\n",
            "bedtools2/src/utils/Point/\n",
            "bedtools2/src/utils/bedGraphFile/\n",
            "bedtools2/src/utils/sequenceUtilities/\n",
            "bedtools2/src/utils/Fasta/\n",
            "bedtools2/src/utils/gzstream/\n",
            "bedtools2/src/utils/fileType/\n",
            "bedtools2/src/utils/KeyListOps/\n",
            "bedtools2/src/utils/BamTools-Ancillary/\n",
            "bedtools2/src/utils/GenomeFile/\n",
            "bedtools2/src/utils/version/\n",
            "bedtools2/src/utils/ToolBase/\n",
            "bedtools2/src/utils/lineFileUtilities/\n",
            "bedtools2/src/utils/bedFilePE/\n",
            "bedtools2/src/utils/chromsweep/\n",
            "bedtools2/src/utils/FileRecordTools/\n",
            "bedtools2/src/utils/BinTree/\n",
            "bedtools2/src/utils/VectorOps/\n",
            "bedtools2/src/utils/stringUtilities/\n",
            "bedtools2/src/utils/stringUtilities/stringUtilities.h\n",
            "bedtools2/src/utils/VectorOps/VectorOps.cpp\n",
            "bedtools2/src/utils/VectorOps/Makefile.frag\n",
            "bedtools2/src/utils/VectorOps/VectorOps.h\n",
            "bedtools2/src/utils/BinTree/BinTree.cpp\n",
            "bedtools2/src/utils/BinTree/BinTree.h\n",
            "bedtools2/src/utils/BinTree/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/Records/\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordList.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/PlusFields.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Record.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Record.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BamRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.h\n",
            "bedtools2/src/utils/chromsweep/chromsweep.cpp\n",
            "bedtools2/src/utils/chromsweep/Makefile.frag\n",
            "bedtools2/src/utils/chromsweep/chromsweep.h\n",
            "bedtools2/src/utils/bedFilePE/bedFilePE.cpp\n",
            "bedtools2/src/utils/bedFilePE/Makefile.frag\n",
            "bedtools2/src/utils/bedFilePE/bedFilePE.h\n",
            "bedtools2/src/utils/lineFileUtilities/lineFileUtilities.h\n",
            "bedtools2/src/utils/ToolBase/ToolBase.cpp\n",
            "bedtools2/src/utils/ToolBase/Makefile.frag\n",
            "bedtools2/src/utils/ToolBase/ToolBase.h\n",
            "bedtools2/src/utils/version/version.h\n",
            "bedtools2/src/utils/version/version.cpp\n",
            "bedtools2/src/utils/version/version_release.txt\n",
            "bedtools2/src/utils/version/Makefile.frag\n",
            "bedtools2/src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "bedtools2/src/utils/GenomeFile/NewGenomeFile.h\n",
            "bedtools2/src/utils/GenomeFile/GenomeFile.cpp\n",
            "bedtools2/src/utils/GenomeFile/Makefile.frag\n",
            "bedtools2/src/utils/GenomeFile/GenomeFile.h\n",
            "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.h\n",
            "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "bedtools2/src/utils/BamTools-Ancillary/Makefile.frag\n",
            "bedtools2/src/utils/KeyListOps/KeyListOps.h\n",
            "bedtools2/src/utils/KeyListOps/KeyListOps.cpp\n",
            "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "bedtools2/src/utils/KeyListOps/Makefile.frag\n",
            "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.h\n",
            "bedtools2/src/utils/fileType/fileType.cpp\n",
            "bedtools2/src/utils/fileType/FileRecordTypeChecker.h\n",
            "bedtools2/src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "bedtools2/src/utils/fileType/Makefile.frag\n",
            "bedtools2/src/utils/fileType/fileType.h\n",
            "bedtools2/src/utils/gzstream/gzstream.C\n",
            "bedtools2/src/utils/gzstream/README\n",
            "bedtools2/src/utils/gzstream/COPYING.LIB\n",
            "bedtools2/src/utils/gzstream/Makefile.frag\n",
            "bedtools2/src/utils/gzstream/version.txt\n",
            "bedtools2/src/utils/gzstream/gzstream.h\n",
            "bedtools2/src/utils/Fasta/Fasta.h\n",
            "bedtools2/src/utils/Fasta/split.cpp\n",
            "bedtools2/src/utils/Fasta/Fasta.cpp\n",
            "bedtools2/src/utils/Fasta/LargeFileSupport.h\n",
            "bedtools2/src/utils/Fasta/Makefile.frag\n",
            "bedtools2/src/utils/Fasta/split.h\n",
            "bedtools2/src/utils/sequenceUtilities/sequenceUtils.h\n",
            "bedtools2/src/utils/sequenceUtilities/Makefile.frag\n",
            "bedtools2/src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "bedtools2/src/utils/bedGraphFile/bedGraphFile.h\n",
            "bedtools2/src/utils/bedGraphFile/Makefile.frag\n",
            "bedtools2/src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "bedtools2/src/utils/Point/Point.h\n",
            "bedtools2/src/utils/driver/BedtoolsDriver.h\n",
            "bedtools2/src/utils/driver/BedtoolsDriver.cpp\n",
            "bedtools2/src/utils/driver/Makefile.frag\n",
            "bedtools2/src/utils/general/QuickString.cpp\n",
            "bedtools2/src/utils/general/ParseTools.cpp\n",
            "bedtools2/src/utils/general/CommonHelp.h\n",
            "bedtools2/src/utils/general/FreeList.h\n",
            "bedtools2/src/utils/general/CompressionTools.cpp\n",
            "bedtools2/src/utils/general/Random.cpp\n",
            "bedtools2/src/utils/general/BedtoolsTypes.h\n",
            "bedtools2/src/utils/general/ErrorMsg.h\n",
            "bedtools2/src/utils/general/PushBackStreamBuf.h\n",
            "bedtools2/src/utils/general/Tokenizer.h\n",
            "bedtools2/src/utils/general/ParseTools.h\n",
            "bedtools2/src/utils/general/QuickString.h\n",
            "bedtools2/src/utils/general/Tokenizer.cpp\n",
            "bedtools2/src/utils/general/ErrorMsg.cpp\n",
            "bedtools2/src/utils/general/CommonHelp.cpp\n",
            "bedtools2/src/utils/general/Makefile.frag\n",
            "bedtools2/src/utils/general/BTlist.h\n",
            "bedtools2/src/utils/general/CompressionTools.h\n",
            "bedtools2/src/utils/general/PushBackStreamBuf.cpp\n",
            "bedtools2/src/utils/general/InflateStreamBuf.h\n",
            "bedtools2/src/utils/general/Random.h\n",
            "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.h\n",
            "bedtools2/src/utils/BlockedIntervals/Makefile.frag\n",
            "bedtools2/src/utils/BamTools/mapping/\n",
            "bedtools2/src/utils/BamTools/include/\n",
            "bedtools2/src/utils/BamTools/.gitignore\n",
            "bedtools2/src/utils/BamTools/Makefile.frag\n",
            "bedtools2/src/utils/BamTools/include/BamAncillary.h\n",
            "bedtools2/src/utils/BamTools/include/SamHeader.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamReader.hpp\n",
            "bedtools2/src/utils/BamTools/include/api/\n",
            "bedtools2/src/utils/BamTools/include/BamWriter.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamAlignment.mapping.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamAlignment.hpp\n",
            "bedtools2/src/utils/BamTools/include/api/BamWriter.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamMultiReader.h\n",
            "bedtools2/src/utils/BamTools/include/api/SamConstants.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamAlignment.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamReader.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamAux.h\n",
            "bedtools2/src/utils/BamTools/mapping/BamAlignment.map\n",
            "bedtools2/src/utils/BamTools/mapping/BamAlignment.py\n",
            "bedtools2/src/utils/Contexts/ContextBase.h\n",
            "bedtools2/src/utils/Contexts/ContextClosest.h\n",
            "bedtools2/src/utils/Contexts/ContextIntersect.h\n",
            "bedtools2/src/utils/Contexts/ContextIntersect.cpp\n",
            "bedtools2/src/utils/Contexts/ContextFisher.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMerge.h\n",
            "bedtools2/src/utils/Contexts/ContextGroupBy.h\n",
            "bedtools2/src/utils/Contexts/ContextComplement.h\n",
            "bedtools2/src/utils/Contexts/ContextSubtract.h\n",
            "bedtools2/src/utils/Contexts/ContextCoverage.cpp\n",
            "bedtools2/src/utils/Contexts/ContextBase.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSummary.h\n",
            "bedtools2/src/utils/Contexts/ContextFisher.h\n",
            "bedtools2/src/utils/Contexts/ContextSpacing.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMap.h\n",
            "bedtools2/src/utils/Contexts/ContextSubtract.cpp\n",
            "bedtools2/src/utils/Contexts/ContextJaccard.h\n",
            "bedtools2/src/utils/Contexts/ContextComplement.cpp\n",
            "bedtools2/src/utils/Contexts/ContextCoverage.h\n",
            "bedtools2/src/utils/Contexts/ContextSample.cpp\n",
            "bedtools2/src/utils/Contexts/ContextClosest.cpp\n",
            "bedtools2/src/utils/Contexts/Makefile.frag\n",
            "bedtools2/src/utils/Contexts/ContextJaccard.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSample.h\n",
            "bedtools2/src/utils/Contexts/ContextSummary.cpp\n",
            "bedtools2/src/utils/Contexts/ContextGroupBy.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMap.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSpacing.h\n",
            "bedtools2/src/utils/Contexts/ContextMerge.cpp\n",
            "bedtools2/src/utils/bedFile/bedFile.h\n",
            "bedtools2/src/utils/bedFile/Makefile.frag\n",
            "bedtools2/src/utils/bedFile/bedFile.cpp\n",
            "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.h\n",
            "bedtools2/src/utils/RecordOutputMgr/Makefile.frag\n",
            "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "bedtools2/src/utils/NewChromsweep/NewChromsweep.h\n",
            "bedtools2/src/utils/NewChromsweep/CloseSweep.h\n",
            "bedtools2/src/utils/NewChromsweep/Makefile.frag\n",
            "bedtools2/src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "bedtools2/src/utils/NewChromsweep/CloseSweep.cpp\n",
            "bedtools2/src/utils/htslib/htslib.mk\n",
            "bedtools2/src/utils/htslib/cram/\n",
            "bedtools2/src/utils/htslib/htslib/\n",
            "bedtools2/src/utils/htslib/faidx.c\n",
            "bedtools2/src/utils/htslib/vcfutils.c\n",
            "bedtools2/src/utils/htslib/bcf_sr_sort.h\n",
            "bedtools2/src/utils/htslib/faidx.5\n",
            "bedtools2/src/utils/htslib/regidx.c\n",
            "bedtools2/src/utils/htslib/htslib.pc.in\n",
            "bedtools2/src/utils/htslib/configure.ac\n",
            "bedtools2/src/utils/htslib/INSTALL\n",
            "bedtools2/src/utils/htslib/hts_os.c\n",
            "bedtools2/src/utils/htslib/tbx.c\n",
            "bedtools2/src/utils/htslib/probaln.c\n",
            "bedtools2/src/utils/htslib/LICENSE\n",
            "bedtools2/src/utils/htslib/kfunc.c\n",
            "bedtools2/src/utils/htslib/hfile_gcs.c\n",
            "bedtools2/src/utils/htslib/errmod.c\n",
            "bedtools2/src/utils/htslib/multipart.c\n",
            "bedtools2/src/utils/htslib/hfile_internal.h\n",
            "bedtools2/src/utils/htslib/Makefile\n",
            "bedtools2/src/utils/htslib/htsfile.c\n",
            "bedtools2/src/utils/htslib/realn.c\n",
            "bedtools2/src/utils/htslib/bgzip.c\n",
            "bedtools2/src/utils/htslib/hfile_net.c\n",
            "bedtools2/src/utils/htslib/tabix.1\n",
            "bedtools2/src/utils/htslib/textutils.c\n",
            "bedtools2/src/utils/htslib/hts.c\n",
            "bedtools2/src/utils/htslib/README\n",
            "bedtools2/src/utils/htslib/vcf_sweep.c\n",
            "bedtools2/src/utils/htslib/README.md\n",
            "bedtools2/src/utils/htslib/sam.5\n",
            "bedtools2/src/utils/htslib/bcf_sr_sort.c\n",
            "bedtools2/src/utils/htslib/synced_bcf_reader.c\n",
            "bedtools2/src/utils/htslib/htslib_vars.mk\n",
            "bedtools2/src/utils/htslib/sam.c\n",
            "bedtools2/src/utils/htslib/NEWS\n",
            "bedtools2/src/utils/htslib/.gitignore\n",
            "bedtools2/src/utils/htslib/vcf.5\n",
            "bedtools2/src/utils/htslib/bgzf.c\n",
            "bedtools2/src/utils/htslib/.gitattributes\n",
            "bedtools2/src/utils/htslib/hts_internal.h\n",
            "bedtools2/src/utils/htslib/vcf.c\n",
            "bedtools2/src/utils/htslib/m4/\n",
            "bedtools2/src/utils/htslib/plugin.c\n",
            "bedtools2/src/utils/htslib/version.sh\n",
            "bedtools2/src/utils/htslib/hfile_libcurl.c\n",
            "bedtools2/src/utils/htslib/kstring.c\n",
            "bedtools2/src/utils/htslib/os/\n",
            "bedtools2/src/utils/htslib/md5.c\n",
            "bedtools2/src/utils/htslib/hfile_s3.c\n",
            "bedtools2/src/utils/htslib/htsfile.1\n",
            "bedtools2/src/utils/htslib/thread_pool.c\n",
            "bedtools2/src/utils/htslib/hfile.c\n",
            "bedtools2/src/utils/htslib/config.mk.in\n",
            "bedtools2/src/utils/htslib/tabix.c\n",
            "bedtools2/src/utils/htslib/textutils_internal.h\n",
            "bedtools2/src/utils/htslib/knetfile.c\n",
            "bedtools2/src/utils/htslib/bgzip.1\n",
            "bedtools2/src/utils/htslib/thread_pool_internal.h\n",
            "bedtools2/src/utils/htslib/os/rand.c\n",
            "bedtools2/src/utils/htslib/os/lzma_stub.h\n",
            "bedtools2/src/utils/htslib/m4/hts_prog_cc_warnings.m4\n",
            "bedtools2/src/utils/htslib/htslib/sam.h\n",
            "bedtools2/src/utils/htslib/htslib/synced_bcf_reader.h\n",
            "bedtools2/src/utils/htslib/htslib/vcf_sweep.h\n",
            "bedtools2/src/utils/htslib/htslib/vcf.h\n",
            "bedtools2/src/utils/htslib/htslib/khash_str2int.h\n",
            "bedtools2/src/utils/htslib/htslib/bgzf.h\n",
            "bedtools2/src/utils/htslib/htslib/cram.h\n",
            "bedtools2/src/utils/htslib/htslib/kstring.h\n",
            "bedtools2/src/utils/htslib/htslib/knetfile.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_log.h\n",
            "bedtools2/src/utils/htslib/htslib/hfile.h\n",
            "bedtools2/src/utils/htslib/htslib/thread_pool.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_defs.h\n",
            "bedtools2/src/utils/htslib/htslib/khash.h\n",
            "bedtools2/src/utils/htslib/htslib/regidx.h\n",
            "bedtools2/src/utils/htslib/htslib/faidx.h\n",
            "bedtools2/src/utils/htslib/htslib/vcfutils.h\n",
            "bedtools2/src/utils/htslib/htslib/kseq.h\n",
            "bedtools2/src/utils/htslib/htslib/kfunc.h\n",
            "bedtools2/src/utils/htslib/htslib/tbx.h\n",
            "bedtools2/src/utils/htslib/htslib/ksort.h\n",
            "bedtools2/src/utils/htslib/htslib/kbitset.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_os.h\n",
            "bedtools2/src/utils/htslib/htslib/klist.h\n",
            "bedtools2/src/utils/htslib/htslib/hts.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_endian.h\n",
            "bedtools2/src/utils/htslib/cram/cram_codecs.h\n",
            "bedtools2/src/utils/htslib/cram/sam_header.c\n",
            "bedtools2/src/utils/htslib/cram/cram_index.c\n",
            "bedtools2/src/utils/htslib/cram/open_trace_file.c\n",
            "bedtools2/src/utils/htslib/cram/mFILE.h\n",
            "bedtools2/src/utils/htslib/cram/cram_external.c\n",
            "bedtools2/src/utils/htslib/cram/cram.h\n",
            "bedtools2/src/utils/htslib/cram/misc.h\n",
            "bedtools2/src/utils/htslib/cram/files.c\n",
            "bedtools2/src/utils/htslib/cram/os.h\n",
            "bedtools2/src/utils/htslib/cram/rANS_static.h\n",
            "bedtools2/src/utils/htslib/cram/cram_samtools.c\n",
            "bedtools2/src/utils/htslib/cram/pooled_alloc.c\n",
            "bedtools2/src/utils/htslib/cram/string_alloc.c\n",
            "bedtools2/src/utils/htslib/cram/cram_io.h\n",
            "bedtools2/src/utils/htslib/cram/cram_decode.h\n",
            "bedtools2/src/utils/htslib/cram/cram_encode.h\n",
            "bedtools2/src/utils/htslib/cram/cram_stats.h\n",
            "bedtools2/src/utils/htslib/cram/mFILE.c\n",
            "bedtools2/src/utils/htslib/cram/cram_index.h\n",
            "bedtools2/src/utils/htslib/cram/sam_header.h\n",
            "bedtools2/src/utils/htslib/cram/open_trace_file.h\n",
            "bedtools2/src/utils/htslib/cram/cram_codecs.c\n",
            "bedtools2/src/utils/htslib/cram/rANS_byte.h\n",
            "bedtools2/src/utils/htslib/cram/string_alloc.h\n",
            "bedtools2/src/utils/htslib/cram/rANS_static.c\n",
            "bedtools2/src/utils/htslib/cram/pooled_alloc.h\n",
            "bedtools2/src/utils/htslib/cram/cram_samtools.h\n",
            "bedtools2/src/utils/htslib/cram/cram_stats.c\n",
            "bedtools2/src/utils/htslib/cram/cram_encode.c\n",
            "bedtools2/src/utils/htslib/cram/cram_structs.h\n",
            "bedtools2/src/utils/htslib/cram/cram_io.c\n",
            "bedtools2/src/utils/htslib/cram/cram_decode.c\n",
            "bedtools2/src/utils/tabFile/tabFile.cpp\n",
            "bedtools2/src/utils/tabFile/tabFile.h\n",
            "bedtools2/src/utils/tabFile/Makefile.frag\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageBed.h\n",
            "bedtools2/src/genomeCoverageBed/Makefile.frag\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBed.h\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "bedtools2/src/maskFastaFromBed/Makefile.frag\n",
            "bedtools2/src/regressTest/RegressTest.h\n",
            "bedtools2/src/regressTest/RegressTest.cpp\n",
            "bedtools2/src/regressTest/regressTestMain.cpp\n",
            "bedtools2/src/regressTest/Makefile.frag\n",
            "bedtools2/src/bedToIgv/bedToIgv.cpp\n",
            "bedtools2/src/bedToIgv/Makefile.frag\n",
            "bedtools2/src/nucBed/nucBed.h\n",
            "bedtools2/src/nucBed/nucBedMain.cpp\n",
            "bedtools2/src/nucBed/LargeFileSupport.h\n",
            "bedtools2/src/nucBed/Makefile.frag\n",
            "bedtools2/src/nucBed/nucBed.cpp\n",
            "bedtools2/src/reldist/reldist.cpp\n",
            "bedtools2/src/reldist/Makefile.frag\n",
            "bedtools2/src/reldist/reldistMain.cpp\n",
            "bedtools2/src/reldist/reldist.h\n",
            "bedtools2/src/getOverlap/getOverlap.cpp\n",
            "bedtools2/src/getOverlap/Makefile.frag\n",
            "bedtools2/src/randomBed/randomBedMain.cpp\n",
            "bedtools2/src/randomBed/randomBed.cpp\n",
            "bedtools2/src/randomBed/Makefile.frag\n",
            "bedtools2/src/randomBed/randomBed.h\n",
            "bedtools2/src/annotateBed/annotateBed.cpp\n",
            "bedtools2/src/annotateBed/annotateBed.h\n",
            "bedtools2/src/annotateBed/annotateMain.cpp\n",
            "bedtools2/src/annotateBed/Makefile.frag\n",
            "bedtools2/src/slopBed/slopBed.h\n",
            "bedtools2/src/slopBed/slopBed.cpp\n",
            "bedtools2/src/slopBed/Makefile.frag\n",
            "bedtools2/src/slopBed/slopBedMain.cpp\n",
            "bedtools2/src/mapFile/mapHelp.cpp\n",
            "bedtools2/src/mapFile/mapFile.cpp\n",
            "bedtools2/src/mapFile/Makefile.frag\n",
            "bedtools2/src/mapFile/mapFile.h\n",
            "bedtools2/src/pairToPair/pairToPair.h\n",
            "bedtools2/src/pairToPair/pairToPairMain.cpp\n",
            "bedtools2/src/pairToPair/pairToPair.cpp\n",
            "bedtools2/src/pairToPair/Makefile.frag\n",
            "bedtools2/src/linksBed/linksBed.cpp\n",
            "bedtools2/src/linksBed/Makefile.frag\n",
            "bedtools2/src/linksBed/linksBed.h\n",
            "bedtools2/src/linksBed/linksMain.cpp\n",
            "bedtools2/src/fisher/fisher.h\n",
            "bedtools2/src/fisher/kfunc.h\n",
            "bedtools2/src/fisher/kfunc.cpp\n",
            "bedtools2/src/fisher/Makefile.frag\n",
            "bedtools2/src/fisher/fisher.cpp\n",
            "bedtools2/src/fisher/fisherHelp.cpp\n",
            "bedtools2/src/expand/Makefile.frag\n",
            "bedtools2/src/expand/expand.cpp\n",
            "bedtools2/src/complementFile/complementHelp.cpp\n",
            "bedtools2/src/complementFile/complementFile.h\n",
            "bedtools2/src/complementFile/Makefile.frag\n",
            "bedtools2/src/complementFile/complementFile.cpp\n",
            "bedtools2/src/bedToBam/bedToBam.cpp\n",
            "bedtools2/src/bedToBam/Makefile.frag\n",
            "bedtools2/src/split/splitBed.h\n",
            "bedtools2/src/split/splitBed.cpp\n",
            "bedtools2/src/split/splitBedMain.cpp\n",
            "bedtools2/src/split/Makefile.frag\n",
            "bedtools2/src/sortBed/sortMain.cpp\n",
            "bedtools2/src/sortBed/sortBed.cpp\n",
            "bedtools2/src/sortBed/sortBed.h\n",
            "bedtools2/src/sortBed/Makefile.frag\n",
            "bedtools2/src/spacingFile/spacingHelp.cpp\n",
            "bedtools2/src/spacingFile/spacingFile.h\n",
            "bedtools2/src/spacingFile/Makefile.frag\n",
            "bedtools2/src/spacingFile/spacingFile.cpp\n",
            "bedtools2/data/gerp.chr1.bed.gz\n",
            "bedtools2/data/simpleRepeats.chr1.bed.gz\n",
            "bedtools2/data/aluY.chr1.bed.gz\n",
            "bedtools2/data/refseq.chr1.exons.bed.gz\n",
            "bedtools2/data/knownGene.hg18.chr21.short.bed\n",
            "bedtools2/data/knownGene.hg18.chr21.bed\n",
            "bedtools2/.github/workflows/\n",
            "bedtools2/.github/workflows/main.yml\n",
            "bedtools2/scripts/makeBashScripts.py\n",
            "bedtools2/scripts/vcfsort.sh\n",
            "bedtools2/genomes/human.hg38.genome\n",
            "bedtools2/genomes/mouse.mm10.genome\n",
            "bedtools2/genomes/mouse.mm8.genome\n",
            "bedtools2/genomes/human.hg18.genome\n",
            "bedtools2/genomes/human.hg19.genome\n",
            "bedtools2/genomes/mouse.mm9.genome\n",
            "bedtools2/docs/index.rst\n",
            "bedtools2/docs/Makefile\n",
            "bedtools2/docs/conf.py\n",
            "bedtools2/docs/_static/\n",
            "bedtools2/docs/content/\n",
            "bedtools2/docs/bedtools.swiss.png\n",
            "bedtools2/docs/build.sh\n",
            "bedtools2/docs/make.bat\n",
            "bedtools2/docs/bedtools.ai\n",
            "bedtools2/docs/templates/\n",
            "bedtools2/docs/themes/\n",
            "bedtools2/docs/bedtools.png\n",
            "bedtools2/docs/themes/rtd/\n",
            "bedtools2/docs/themes/rtd/theme.conf\n",
            "bedtools2/docs/themes/rtd/static/\n",
            "bedtools2/docs/themes/rtd/static/rtd.css_t\n",
            "bedtools2/docs/templates/page.html\n",
            "bedtools2/docs/templates/sidebar-intro.html\n",
            "bedtools2/docs/content/general-usage.rst\n",
            "bedtools2/docs/content/tools/\n",
            "bedtools2/docs/content/images/\n",
            "bedtools2/docs/content/quick-start.rst\n",
            "bedtools2/docs/content/example-usage.rst\n",
            "bedtools2/docs/content/overview.rst\n",
            "bedtools2/docs/content/advanced-usage.rst\n",
            "bedtools2/docs/content/history.rst\n",
            "bedtools2/docs/content/installation.rst\n",
            "bedtools2/docs/content/tips-and-tricks.rst\n",
            "bedtools2/docs/content/bedtools-suite.rst\n",
            "bedtools2/docs/content/faq.rst\n",
            "bedtools2/docs/content/related-tools.rst\n",
            "bedtools2/docs/content/images/memory-comparo.png\n",
            "bedtools2/docs/content/images/speed-comparo.png\n",
            "bedtools2/docs/content/images/fisher/\n",
            "bedtools2/docs/content/images/github-zip-button.png\n",
            "bedtools2/docs/content/images/galaxy-bedtools.png\n",
            "bedtools2/docs/content/images/bedtoolsMenu.png\n",
            "bedtools2/docs/content/images/tool-glyphs/\n",
            "bedtools2/docs/content/images/tool-glyphs/map-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/jaccard-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/merge-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/getfasta-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/map-speed-comparo.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-plot.png\n",
            "bedtools2/docs/content/images/tool-glyphs/subtract-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/flank-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/jaccard-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/map-glyph.svg\n",
            "bedtools2/docs/content/images/tool-glyphs/random-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/genomecov-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/cluster-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/maskfasta-glyph.svg\n",
            "bedtools2/docs/content/images/tool-glyphs/random-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/complement-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/slop-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/window-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/shuffle-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/closest-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/maskfasta-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/intersect-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/barski_binding_site.png\n",
            "bedtools2/docs/content/images/fisher/res-genes.png\n",
            "bedtools2/docs/content/tools/expand.rst\n",
            "bedtools2/docs/content/tools/genomecov.rst\n",
            "bedtools2/docs/content/tools/makewindows.rst\n",
            "bedtools2/docs/content/tools/igv.rst\n",
            "bedtools2/docs/content/tools/overlap.rst\n",
            "bedtools2/docs/content/tools/nuc.rst\n",
            "bedtools2/docs/content/tools/fisher.rst\n",
            "bedtools2/docs/content/tools/cluster.rst\n",
            "bedtools2/docs/content/tools/pairtobed.rst\n",
            "bedtools2/docs/content/tools/shift.rst\n",
            "bedtools2/docs/content/tools/complement.rst\n",
            "bedtools2/docs/content/tools/bed12tobed6.rst\n",
            "bedtools2/docs/content/tools/map.rst\n",
            "bedtools2/docs/content/tools/tag.rst\n",
            "bedtools2/docs/content/tools/pairtopair.rst\n",
            "bedtools2/docs/content/tools/slop.rst\n",
            "bedtools2/docs/content/tools/sort.rst\n",
            "bedtools2/docs/content/tools/reldist.rst\n",
            "bedtools2/docs/content/tools/closest.rst\n",
            "bedtools2/docs/content/tools/bedtobam.rst\n",
            "bedtools2/docs/content/tools/maskfasta.rst\n",
            "bedtools2/docs/content/tools/bamtofastq.rst\n",
            "bedtools2/docs/content/tools/multicov.rst\n",
            "bedtools2/docs/content/tools/flank.rst\n",
            "bedtools2/docs/content/tools/random.rst\n",
            "bedtools2/docs/content/tools/merge.rst\n",
            "bedtools2/docs/content/tools/shuffle.rst\n",
            "bedtools2/docs/content/tools/window.rst\n",
            "bedtools2/docs/content/tools/bedpetobam.rst\n",
            "bedtools2/docs/content/tools/groupby.rst\n",
            "bedtools2/docs/content/tools/multiinter.rst\n",
            "bedtools2/docs/content/tools/bamtobed.rst\n",
            "bedtools2/docs/content/tools/getfasta.rst\n",
            "bedtools2/docs/content/tools/unionbedg.rst\n",
            "bedtools2/docs/content/tools/subtract.rst\n",
            "bedtools2/docs/content/tools/coverage.rst\n",
            "bedtools2/docs/content/tools/jaccard.rst\n",
            "bedtools2/docs/content/tools/links.rst\n",
            "bedtools2/docs/content/tools/annotate.rst\n",
            "bedtools2/docs/content/tools/intersect.rst\n",
            "bedtools2/docs/_static/labibi.css\n",
            "bedtools2/docs/_static/labibi.js\n",
            "bedtools2/test/split/\n",
            "bedtools2/test/bedtobam/\n",
            "bedtools2/test/cluster/\n",
            "bedtools2/test/expand/\n",
            "bedtools2/test/shuffle/\n",
            "bedtools2/test/fisher/\n",
            "bedtools2/test/complement/\n",
            "bedtools2/test/intersect/\n",
            "bedtools2/test/sort/\n",
            "bedtools2/test/subtract/\n",
            "bedtools2/test/merge/\n",
            "bedtools2/test/general/\n",
            "bedtools2/test/slop/\n",
            "bedtools2/test/reldist/\n",
            "bedtools2/test/shift/\n",
            "bedtools2/test/genomecov/\n",
            "bedtools2/test/jaccard/\n",
            "bedtools2/test/groupby/\n",
            "bedtools2/test/bamtofastq/\n",
            "bedtools2/test/bed12tobed6/\n",
            "bedtools2/test/getfasta/\n",
            "bedtools2/test/map/\n",
            "bedtools2/test/bigchroms/\n",
            "bedtools2/test/spacing/\n",
            "bedtools2/test/sample/\n",
            "bedtools2/test/flank/\n",
            "bedtools2/test/negativecontrol/\n",
            "bedtools2/test/multicov/\n",
            "bedtools2/test/closest/\n",
            "bedtools2/test/test.sh\n",
            "bedtools2/test/coverage/\n",
            "bedtools2/test/bamtobed/\n",
            "bedtools2/test/makewindows/\n",
            "bedtools2/test/makewindows/test-makewindows.sh\n",
            "bedtools2/test/makewindows/a.33bp.bed\n",
            "bedtools2/test/makewindows/a.20bp.bed\n",
            "bedtools2/test/makewindows/input.bed\n",
            "bedtools2/test/makewindows/a.19bp.bed\n",
            "bedtools2/test/bamtobed/two_blocks.sam\n",
            "bedtools2/test/bamtobed/three_blocks.sam\n",
            "bedtools2/test/bamtobed/test-bamtobed.sh\n",
            "bedtools2/test/bamtobed/numeric_tag.sam\n",
            "bedtools2/test/bamtobed/extra-long-header.sam.gz\n",
            "bedtools2/test/bamtobed/two_blocks_w_D.sam\n",
            "bedtools2/test/bamtobed/one_block.sam\n",
            "bedtools2/test/coverage/two_blocks.sam\n",
            "bedtools2/test/coverage/chr_0-100.bed\n",
            "bedtools2/test/coverage/three_blocks.bam.bai\n",
            "bedtools2/test/coverage/sam-w-del.sam\n",
            "bedtools2/test/coverage/three_blocks_nomatch.bed\n",
            "bedtools2/test/coverage/two_blocks.bam.bai\n",
            "bedtools2/test/coverage/three_blocks.sam\n",
            "bedtools2/test/coverage/one_block.bam.bai\n",
            "bedtools2/test/coverage/test-coverage.sh\n",
            "bedtools2/test/coverage/x.bed\n",
            "bedtools2/test/coverage/y.bed\n",
            "bedtools2/test/coverage/a.bed\n",
            "bedtools2/test/coverage/c.bed\n",
            "bedtools2/test/coverage/three_blocks_match.bam\n",
            "bedtools2/test/coverage/sam-w-del.bam.bai\n",
            "bedtools2/test/coverage/b.bed\n",
            "bedtools2/test/coverage/three_blocks_match.bed\n",
            "bedtools2/test/coverage/one_block.sam\n",
            "bedtools2/test/coverage/three_blocks_match_1bp.bed\n",
            "bedtools2/test/closest/d_d1F.bed\n",
            "bedtools2/test/closest/d_q1.bed\n",
            "bedtools2/test/closest/d_iu.bed\n",
            "bedtools2/test/closest/d_d1R.bed\n",
            "bedtools2/test/closest/d_q2.bed\n",
            "bedtools2/test/closest/tfbs.bed\n",
            "bedtools2/test/closest/null_b.bed\n",
            "bedtools2/test/closest/bug281_a.medium.bed\n",
            "bedtools2/test/closest/null_c.bed\n",
            "bedtools2/test/closest/null_a.bed\n",
            "bedtools2/test/closest/d_id.bed\n",
            "bedtools2/test/closest/a2.bed\n",
            "bedtools2/test/closest/test-closest.sh\n",
            "bedtools2/test/closest/kclosest/\n",
            "bedtools2/test/closest/close-a.bed\n",
            "bedtools2/test/closest/b.names.bed\n",
            "bedtools2/test/closest/close-b.bed\n",
            "bedtools2/test/closest/strand-test-d.bed\n",
            "bedtools2/test/closest/b-one-bp-closer.bed\n",
            "bedtools2/test/closest/islands.2.bed\n",
            "bedtools2/test/closest/bug157_b.bed\n",
            "bedtools2/test/closest/shores.bed\n",
            "bedtools2/test/closest/strand-test-a.bed\n",
            "bedtools2/test/closest/strand-test-c.bed\n",
            "bedtools2/test/closest/bug157_a.bed\n",
            "bedtools2/test/closest/strand-test-b.bed\n",
            "bedtools2/test/closest/bug281_b.bed\n",
            "bedtools2/test/closest/dmr.bed\n",
            "bedtools2/test/closest/a.bed\n",
            "bedtools2/test/closest/b.bed\n",
            "bedtools2/test/closest/a.names.bed\n",
            "bedtools2/test/closest/mdb3.bed\n",
            "bedtools2/test/closest/mdb2.bed\n",
            "bedtools2/test/closest/b2.bed\n",
            "bedtools2/test/closest/mdb1.bed\n",
            "bedtools2/test/closest/mq1.bed\n",
            "bedtools2/test/closest/d.bed\n",
            "bedtools2/test/closest/d4.bed\n",
            "bedtools2/test/closest/sortAndNaming/\n",
            "bedtools2/test/closest/islands.bed\n",
            "bedtools2/test/closest/bug281_b.flip.bed\n",
            "bedtools2/test/closest/bug281_a.flip.medium.bed\n",
            "bedtools2/test/closest/d_d2R.bed\n",
            "bedtools2/test/closest/d_d2F.bed\n",
            "bedtools2/test/closest/sortAndNaming/num_all.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_noChr.bed\n",
            "bedtools2/test/closest/sortAndNaming/alpha_all.bed\n",
            "bedtools2/test/closest/sortAndNaming/test-sort-and-naming.sh\n",
            "bedtools2/test/closest/sortAndNaming/q1_gls.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db2_numBackwards.bed\n",
            "bedtools2/test/closest/sortAndNaming/num_missing.bed\n",
            "bedtools2/test/closest/sortAndNaming/q1_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db2_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_leadingZero.txt\n",
            "bedtools2/test/closest/sortAndNaming/sq1.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_gls.bed\n",
            "bedtools2/test/closest/sortAndNaming/alpha_missing.bed\n",
            "bedtools2/test/closest/sortAndNaming/sdb1.bed\n",
            "bedtools2/test/closest/sortAndNaming/db3_numBackwards.bed\n",
            "bedtools2/test/closest/sortAndNaming/q1a_num.bed\n",
            "bedtools2/test/closest/kclosest/test-kclosest.sh\n",
            "bedtools2/test/closest/kclosest/bug471_b.bed\n",
            "bedtools2/test/closest/kclosest/bug471_a.bed\n",
            "bedtools2/test/closest/kclosest/d2.bed\n",
            "bedtools2/test/closest/kclosest/q2.bed\n",
            "bedtools2/test/closest/kclosest/d3.bed\n",
            "bedtools2/test/closest/kclosest/d1.bed\n",
            "bedtools2/test/closest/kclosest/q1.bed\n",
            "bedtools2/test/multicov/two_blocks.sam\n",
            "bedtools2/test/multicov/test-multicov.sh\n",
            "bedtools2/test/multicov/test-multi.bed\n",
            "bedtools2/test/multicov/test-multi.2.sam\n",
            "bedtools2/test/multicov/multicov.bed\n",
            "bedtools2/test/multicov/test-multi.sam\n",
            "bedtools2/test/multicov/one_block.sam\n",
            "bedtools2/test/negativecontrol/test-negativecontrol.sh\n",
            "bedtools2/test/flank/test-flank.sh\n",
            "bedtools2/test/flank/a.bed\n",
            "bedtools2/test/flank/tiny.genome\n",
            "bedtools2/test/sample/test-sample.sh\n",
            "bedtools2/test/sample/human.hg19.genome\n",
            "bedtools2/test/spacing/a.bed\n",
            "bedtools2/test/spacing/test-spacing.sh\n",
            "bedtools2/test/bigchroms/test-bigchroms.sh\n",
            "bedtools2/test/bigchroms/big4.bed\n",
            "bedtools2/test/bigchroms/make-big-chrom.py\n",
            "bedtools2/test/bigchroms/big4c.bed\n",
            "bedtools2/test/bigchroms/bigx.bed\n",
            "bedtools2/test/bigchroms/abig.bed\n",
            "bedtools2/test/map/values.bam\n",
            "bedtools2/test/map/.Rhistory\n",
            "bedtools2/test/map/a.vsorted.bed\n",
            "bedtools2/test/map/three_blocks_nomatch.bed\n",
            "bedtools2/test/map/b.vsorted.bed\n",
            "bedtools2/test/map/test.vcf\n",
            "bedtools2/test/map/values.bed\n",
            "bedtools2/test/map/test.gff\n",
            "bedtools2/test/map/values5.bed\n",
            "bedtools2/test/map/ivls.bed\n",
            "bedtools2/test/map/values4.bed\n",
            "bedtools2/test/map/t.bed\n",
            "bedtools2/test/map/ivls2.bed\n",
            "bedtools2/test/map/values3.bed\n",
            "bedtools2/test/map/.RData\n",
            "bedtools2/test/map/values2.bed\n",
            "bedtools2/test/map/d.bed\n",
            "bedtools2/test/map/genome\n",
            "bedtools2/test/map/three_blocks_match.bed\n",
            "bedtools2/test/map/test-map.sh\n",
            "bedtools2/test/map/bug262_a.bed\n",
            "bedtools2/test/map/bug262_b.bed\n",
            "bedtools2/test/map/fullFields.bam\n",
            "bedtools2/test/getfasta/test.iupac.bed\n",
            "bedtools2/test/getfasta/rna.fasta\n",
            "bedtools2/test/getfasta/t.fa\n",
            "bedtools2/test/getfasta/test.iupac.fa\n",
            "bedtools2/test/getfasta/blocks.bed\n",
            "bedtools2/test/getfasta/t_fH.fa.fai\n",
            "bedtools2/test/getfasta/t.bed\n",
            "bedtools2/test/getfasta/t.fa.fai\n",
            "bedtools2/test/getfasta/test-getfasta.sh\n",
            "bedtools2/test/getfasta/t_fH.fa\n",
            "bedtools2/test/getfasta/rna.bed\n",
            "bedtools2/test/bed12tobed6/two_blocks.bed\n",
            "bedtools2/test/bed12tobed6/bed6.bed\n",
            "bedtools2/test/bed12tobed6/three_blocks.bed\n",
            "bedtools2/test/bed12tobed6/test-bed12tobed6.sh\n",
            "bedtools2/test/bed12tobed6/one_blocks.bed\n",
            "bedtools2/test/bamtofastq/test-bamtofastq.sh\n",
            "bedtools2/test/bamtofastq/test2.sam\n",
            "bedtools2/test/bamtofastq/golden.fq\n",
            "bedtools2/test/bamtofastq/golden.fq2\n",
            "bedtools2/test/bamtofastq/test.sam\n",
            "bedtools2/test/groupby/gdc.bam\n",
            "bedtools2/test/groupby/values3_case.header.bed\n",
            "bedtools2/test/groupby/test-groupby.sh\n",
            "bedtools2/test/groupby/values3.unmarked_header.bed\n",
            "bedtools2/test/groupby/a_vcfSVtest.vcf\n",
            "bedtools2/test/groupby/values3.7fields.header.bed\n",
            "bedtools2/test/groupby/values3.unmarked_header.bed.2\n",
            "bedtools2/test/groupby/values3.no_header.bed\n",
            "bedtools2/test/groupby/values3.header.bed\n",
            "bedtools2/test/groupby/noPosvalues.header.bed\n",
            "bedtools2/test/groupby/test.bed\n",
            "bedtools2/test/groupby/bug569_problem.txt\n",
            "bedtools2/test/jaccard/a645.bed\n",
            "bedtools2/test/jaccard/short.bed\n",
            "bedtools2/test/jaccard/bMixedStrands.bed\n",
            "bedtools2/test/jaccard/a.bed\n",
            "bedtools2/test/jaccard/c.bed\n",
            "bedtools2/test/jaccard/three_blocks_match.bam\n",
            "bedtools2/test/jaccard/b.bed\n",
            "bedtools2/test/jaccard/b645.bed\n",
            "bedtools2/test/jaccard/e.bed\n",
            "bedtools2/test/jaccard/test-jaccard.sh\n",
            "bedtools2/test/jaccard/long.bed\n",
            "bedtools2/test/jaccard/three_blocks_match.bed\n",
            "bedtools2/test/jaccard/aMixedStrands.bed\n",
            "bedtools2/test/jaccard/a.bam\n",
            "bedtools2/test/genomecov/two_blocks.sam\n",
            "bedtools2/test/genomecov/test_ref.fa.fai\n",
            "bedtools2/test/genomecov/test-genomecov.sh\n",
            "bedtools2/test/genomecov/chip.sam\n",
            "bedtools2/test/genomecov/y.bam\n",
            "bedtools2/test/genomecov/test_ref.fa\n",
            "bedtools2/test/genomecov/sam-w-del.sam\n",
            "bedtools2/test/genomecov/three_blocks_nomatch.bed\n",
            "bedtools2/test/genomecov/three_blocks.sam\n",
            "bedtools2/test/genomecov/empty.bam\n",
            "bedtools2/test/genomecov/y.bed\n",
            "bedtools2/test/genomecov/empty.cram\n",
            "bedtools2/test/genomecov/genome.txt\n",
            "bedtools2/test/genomecov/three_blocks_match.bed\n",
            "bedtools2/test/genomecov/pair-chip.sam\n",
            "bedtools2/test/genomecov/one_block.sam\n",
            "bedtools2/test/genomecov/three_blocks_match_1bp.bed\n",
            "bedtools2/test/shift/test-shift.sh\n",
            "bedtools2/test/shift/issue_807.bed\n",
            "bedtools2/test/shift/a.bed\n",
            "bedtools2/test/shift/b.bed\n",
            "bedtools2/test/shift/tiny.genome\n",
            "bedtools2/test/shift/huge.genome\n",
            "bedtools2/test/shift/issue_807.genomesize\n",
            "bedtools2/test/reldist/issue_711.a.bed\n",
            "bedtools2/test/reldist/issue_711.b.bed\n",
            "bedtools2/test/reldist/test-reldist.sh\n",
            "bedtools2/test/slop/a.bed\n",
            "bedtools2/test/slop/tiny.genome\n",
            "bedtools2/test/slop/test-slop.sh\n",
            "bedtools2/test/general/hg19.fa.fai\n",
            "bedtools2/test/general/empty.bed\n",
            "bedtools2/test/general/a.chromheader.bed\n",
            "bedtools2/test/general/test-general.sh\n",
            "bedtools2/test/general/non-empty.bed.gz\n",
            "bedtools2/test/general/t.bed\n",
            "bedtools2/test/general/a.trackheader.bed\n",
            "bedtools2/test/merge/bamCol8Mean.txt\n",
            "bedtools2/test/merge/mergeBugTest.bed\n",
            "bedtools2/test/merge/expFormat.bed\n",
            "bedtools2/test/merge/bamCol9Mean.txt\n",
            "bedtools2/test/merge/unsorted.bed\n",
            "bedtools2/test/merge/bug254_d.bed\n",
            "bedtools2/test/merge/bug254_e.bed\n",
            "bedtools2/test/merge/bamCol6Collapse.txt\n",
            "bedtools2/test/merge/bamCol3Collapse.txt\n",
            "bedtools2/test/merge/vcfSVtest.vcf\n",
            "bedtools2/test/merge/bug254_c.bed\n",
            "bedtools2/test/merge/bamCol11Collapse.txt\n",
            "bedtools2/test/merge/a.full.bed\n",
            "bedtools2/test/merge/a.bed\n",
            "bedtools2/test/merge/test-merge.sh\n",
            "bedtools2/test/merge/precisionTest.bed\n",
            "bedtools2/test/merge/b.bed\n",
            "bedtools2/test/merge/a.names.bed\n",
            "bedtools2/test/merge/testA.vcf\n",
            "bedtools2/test/merge/bamCol1Collapse.txt\n",
            "bedtools2/test/merge/precisionTest2.bed\n",
            "bedtools2/test/merge/bamCol10Collapse.txt\n",
            "bedtools2/test/merge/bamCol5Mean.txt\n",
            "bedtools2/test/merge/a.gff\n",
            "bedtools2/test/merge/bamCol7Collapse.txt\n",
            "bedtools2/test/merge/bamCol4Mean.txt\n",
            "bedtools2/test/merge/a.full.bam\n",
            "bedtools2/test/merge/mixedStrands.bed\n",
            "bedtools2/test/merge/fullFields.bam\n",
            "bedtools2/test/subtract/test-subtract.sh\n",
            "bedtools2/test/subtract/a.bed\n",
            "bedtools2/test/subtract/c.bed\n",
            "bedtools2/test/subtract/b.bed\n",
            "bedtools2/test/subtract/b2.bed\n",
            "bedtools2/test/subtract/d.bed\n",
            "bedtools2/test/sort/names.txt\n",
            "bedtools2/test/sort/a.bed\n",
            "bedtools2/test/sort/test-sort.sh\n",
            "bedtools2/test/sort/b.bed\n",
            "bedtools2/test/intersect/two_blocks.sam\n",
            "bedtools2/test/intersect/a_with_bothUnmapped.bam\n",
            "bedtools2/test/intersect/gdc.gff\n",
            "bedtools2/test/intersect/bug223_sv1_b.vcf\n",
            "bedtools2/test/intersect/bed3.bed\n",
            "bedtools2/test/intersect/bug150_a.bed\n",
            "bedtools2/test/intersect/test_ref.fa.fai\n",
            "bedtools2/test/intersect/x.bam\n",
            "bedtools2/test/intersect/exons.issue750.bed\n",
            "bedtools2/test/intersect/performanceTest.sh\n",
            "bedtools2/test/intersect/bed6.strand.bed\n",
            "bedtools2/test/intersect/queryUnmappedMateMappedCoordsInvalid.bam\n",
            "bedtools2/test/intersect/bug223_sv1_a.vcf\n",
            "bedtools2/test/intersect/bug150_b.bed\n",
            "bedtools2/test/intersect/headerOnly.vcf\n",
            "bedtools2/test/intersect/gdc_one.bed\n",
            "bedtools2/test/intersect/bed6.strand2.bed\n",
            "bedtools2/test/intersect/null_b.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader.bed\n",
            "bedtools2/test/intersect/human.hg19.vSort.genome\n",
            "bedtools2/test/intersect/gdc.bam\n",
            "bedtools2/test/intersect/bed4.bed\n",
            "bedtools2/test/intersect/chromOrderB.bed\n",
            "bedtools2/test/intersect/bedplus.bed\n",
            "bedtools2/test/intersect/test_ref.fa\n",
            "bedtools2/test/intersect/bed5.bed\n",
            "bedtools2/test/intersect/oneRecordNoNewline.bed\n",
            "bedtools2/test/intersect/aWithHeaderVsB.txt\n",
            "bedtools2/test/intersect/null_c.bed\n",
            "bedtools2/test/intersect/j1.bed\n",
            "bedtools2/test/intersect/b.issue311.vcf\n",
            "bedtools2/test/intersect/null_a.bed\n",
            "bedtools2/test/intersect/oneUnmapped.bam\n",
            "bedtools2/test/intersect/chromOrderA.bed\n",
            "bedtools2/test/intersect/bed6.bed\n",
            "bedtools2/test/intersect/issue_773_b.bed\n",
            "bedtools2/test/intersect/new_test-intersect.sh\n",
            "bedtools2/test/intersect/chromsOutOfOrder.bed\n",
            "bedtools2/test/intersect/jim.bed\n",
            "bedtools2/test/intersect/three_blocks_nomatch.bed\n",
            "bedtools2/test/intersect/b_vcfSVtest.vcf\n",
            "bedtools2/test/intersect/bug223_f.vcf\n",
            "bedtools2/test/intersect/a.issue311.bed\n",
            "bedtools2/test/intersect/three_blocks.sam\n",
            "bedtools2/test/intersect/gdc_exp\n",
            "bedtools2/test/intersect/bug44_b.bed\n",
            "bedtools2/test/intersect/bug223_d.vcf\n",
            "bedtools2/test/intersect/bug223_e.vcf\n",
            "bedtools2/test/intersect/a_testZeroLen.bed\n",
            "bedtools2/test/intersect/a_bgzipped.bed.gz\n",
            "bedtools2/test/intersect/x.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader_bgzipped.bed.gz\n",
            "bedtools2/test/intersect/dosLineChar_a.bed\n",
            "bedtools2/test/intersect/bug44_a.vcf.gz\n",
            "bedtools2/test/intersect/b.cram\n",
            "bedtools2/test/intersect/y.bed\n",
            "bedtools2/test/intersect/test-intersect.sh\n",
            "bedtools2/test/intersect/a_vcfSVtest.vcf\n",
            "bedtools2/test/intersect/blocks.bed12\n",
            "bedtools2/test/intersect/recordsOutOfOrder.bed\n",
            "bedtools2/test/intersect/a.bed\n",
            "bedtools2/test/intersect/jim.vcf\n",
            "bedtools2/test/intersect/c.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bam\n",
            "bedtools2/test/intersect/dosLineCharWithExtraTab_b.bed\n",
            "bedtools2/test/intersect/a.cram\n",
            "bedtools2/test/intersect/b.bed\n",
            "bedtools2/test/intersect/two_blocks_partial.bam.bai\n",
            "bedtools2/test/intersect/foo\n",
            "bedtools2/test/intersect/split.issue750.bam\n",
            "bedtools2/test/intersect/mapped_and_unmapped.sam\n",
            "bedtools2/test/intersect/two_blocks_partial.bam\n",
            "bedtools2/test/intersect/e.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader_gzipped.bed.gz\n",
            "bedtools2/test/intersect/svlen.vcf\n",
            "bedtools2/test/intersect/d.bed\n",
            "bedtools2/test/intersect/splitBug155_a.bed\n",
            "bedtools2/test/intersect/multi_intersect/\n",
            "bedtools2/test/intersect/notexthdr.bam\n",
            "bedtools2/test/intersect/large_b.bed\n",
            "bedtools2/test/intersect/nonamecheck_a.bed\n",
            "bedtools2/test/intersect/issue_773_y.bed\n",
            "bedtools2/test/intersect/sortAndNaming/\n",
            "bedtools2/test/intersect/split.issue750.sam\n",
            "bedtools2/test/intersect/a_gzipped.bed.gz\n",
            "bedtools2/test/intersect/splitBug155_b.bed\n",
            "bedtools2/test/intersect/bug167_strandSweep.bed\n",
            "bedtools2/test/intersect/large_a.bed\n",
            "bedtools2/test/intersect/nonamecheck_b.bed\n",
            "bedtools2/test/intersect/b.issue311.gff\n",
            "bedtools2/test/intersect/two_blocks_partial.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bed\n",
            "bedtools2/test/intersect/human.hg19.genome\n",
            "bedtools2/test/intersect/bug187_a.bed\n",
            "bedtools2/test/intersect/strand_with_star.b.bed\n",
            "bedtools2/test/intersect/b_testZeroLen.bed\n",
            "bedtools2/test/intersect/one_block.sam\n",
            "bedtools2/test/intersect/bug187_b.bed\n",
            "bedtools2/test/intersect/three_blocks_match_1bp.bed\n",
            "bedtools2/test/intersect/strand_with_star.a.bed\n",
            "bedtools2/test/intersect/aVSb.bam\n",
            "bedtools2/test/intersect/a.bam\n",
            "bedtools2/test/intersect/sortAndNaming/num_all.bed\n",
            "bedtools2/test/intersect/sortAndNaming/diffs.txt\n",
            "bedtools2/test/intersect/sortAndNaming/db1_noChr.bed\n",
            "bedtools2/test/intersect/sortAndNaming/alpha_all.bed\n",
            "bedtools2/test/intersect/sortAndNaming/test-sort-and-naming.sh\n",
            "bedtools2/test/intersect/sortAndNaming/exp1\n",
            "bedtools2/test/intersect/sortAndNaming/q1_gls.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db2_numBackwards.bed\n",
            "bedtools2/test/intersect/sortAndNaming/num_missing.bed\n",
            "bedtools2/test/intersect/sortAndNaming/q1_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db2_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_leadingZero.txt\n",
            "bedtools2/test/intersect/sortAndNaming/sq1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_gls.bed\n",
            "bedtools2/test/intersect/sortAndNaming/alpha_missing.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/\n",
            "bedtools2/test/intersect/sortAndNaming/sdb1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db3_numBackwards.bed\n",
            "bedtools2/test/intersect/sortAndNaming/q1a_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/curr5v5.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_500Kvs500K.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/q500K.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/dbchr1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_chr1.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/diffs.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/curr500Kvs500K.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/qchr1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_5v5.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/currChr1.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/db500K.bed\n",
            "bedtools2/test/intersect/multi_intersect/query3.bed\n",
            "bedtools2/test/intersect/multi_intersect/query2.bed\n",
            "bedtools2/test/intersect/multi_intersect/d8.bed\n",
            "bedtools2/test/intersect/multi_intersect/d9.bed\n",
            "bedtools2/test/intersect/multi_intersect/query.bed\n",
            "bedtools2/test/intersect/multi_intersect/g.bed\n",
            "bedtools2/test/intersect/multi_intersect/test-multi_intersect.sh\n",
            "bedtools2/test/intersect/multi_intersect/d4.bed\n",
            "bedtools2/test/intersect/multi_intersect/d5.bed\n",
            "bedtools2/test/intersect/multi_intersect/d7.bed\n",
            "bedtools2/test/intersect/multi_intersect/d6.bed\n",
            "bedtools2/test/intersect/multi_intersect/d2.bed\n",
            "bedtools2/test/intersect/multi_intersect/d3.bed\n",
            "bedtools2/test/intersect/multi_intersect/d1.bed\n",
            "bedtools2/test/complement/test-complement.sh\n",
            "bedtools2/test/complement/issue_503.genome\n",
            "bedtools2/test/complement/issue_503.bed\n",
            "bedtools2/test/fisher/test-fisher.sh\n",
            "bedtools2/test/fisher/sim.py\n",
            "bedtools2/test/fisher/plot.py\n",
            "bedtools2/test/fisher/cmp.sh\n",
            "bedtools2/test/fisher/dm6.fai\n",
            "bedtools2/test/fisher/t.60.genome\n",
            "bedtools2/test/fisher/get.sh\n",
            "bedtools2/test/fisher/README.md\n",
            "bedtools2/test/fisher/a.bed\n",
            "bedtools2/test/fisher/b.bed\n",
            "bedtools2/test/fisher/tumor.gff\n",
            "bedtools2/test/fisher/t.500.genome\n",
            "bedtools2/test/fisher/a_merge.bed\n",
            "bedtools2/test/fisher/shuf.sh\n",
            "bedtools2/test/fisher/test.bed\n",
            "bedtools2/test/shuffle/incl.sizedif.bed\n",
            "bedtools2/test/shuffle/test-shuffle.sh\n",
            "bedtools2/test/shuffle/excl.bed\n",
            "bedtools2/test/shuffle/simrep.bed\n",
            "bedtools2/test/shuffle/incl.bed\n",
            "bedtools2/test/expand/expand.txt\n",
            "bedtools2/test/expand/test-expand.sh\n",
            "bedtools2/test/cluster/in.bed\n",
            "bedtools2/test/cluster/test-cluster.sh\n",
            "bedtools2/test/bedtobam/test-bedtobam.sh\n",
            "bedtools2/test/bedtobam/chrsize.tmp\n",
            "bedtools2/test/split/test-split.sh\n",
            "bedtools2/test/split/randData.bed\n",
            "bedtools2/tutorial/answers.md\n",
            "bedtools2/tutorial/template.class.html\n",
            "bedtools2/tutorial/answers.html\n",
            "bedtools2/tutorial/bootstrap.css\n",
            "bedtools2/tutorial/Makefile\n",
            "bedtools2/tutorial/bedtools.html\n",
            "bedtools2/tutorial/bedtools.md\n",
            "bedtools2/tutorial/template.html\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cd bedtools2; make"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zOS8fxt4Gf00",
        "outputId": "02e3d518-4bfd-493d-cd38-c8e3c1a1a63b"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Building BEDTools:\n",
            "=========================================================\n",
            "CXXFLAGS is [-g -Wall -O2 -std=c++11 ]\n",
            "DETECTED_VERSION = v2.30.0\n",
            "CURRENT_VERSION  = \n",
            "Updating version file.\n",
            "  * compiling src/bedtools.cpp\n",
            "  * compiling src/annotateBed/annotateMain.cpp\n",
            "  * compiling src/annotateBed/annotateBed.cpp\n",
            "  * compiling src/bamToBed/bamToBed.cpp\n",
            "\u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bamtobed_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:88:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KuseAlignmentScore\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   88 |     bool \u001b[01;35m\u001b[KuseAlignmentScore\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bamToFastq/bamToFastqMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/bamToFastq/bamToFastqMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bamtofastq_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bamToFastq/bamToFastqMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFastq2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveFastq2\u001b[m\u001b[K    = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bamToFastq/bamToFastq.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToFastq/bamToFastq.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToFastq/bamToFastq.cpp:15\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedToBam/bedToBam.cpp\n",
            "\u001b[01m\u001b[Ksrc/bedToBam/bedToBam.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bedtobam_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bedToBam/bedToBam.cpp:60:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMapQual\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   60 |     bool \u001b[01;35m\u001b[KhaveMapQual\u001b[m\u001b[K     = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedpeToBam/bedpeToBam.cpp\n",
            "\u001b[01m\u001b[Ksrc/bedpeToBam/bedpeToBam.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bedpetobam_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bedpeToBam/bedpeToBam.cpp:61:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMapQual\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   61 |     bool \u001b[01;35m\u001b[KhaveMapQual\u001b[m\u001b[K     = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedToIgv/bedToIgv.cpp\n",
            "  * compiling src/bed12ToBed6/bed12ToBed6.cpp\n",
            "  * compiling src/closestFile/closestHelp.cpp\n",
            "  * compiling src/closestFile/closestFile.cpp\n",
            "  * compiling src/clusterBed/clusterMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/clusterBed/clusterMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint cluster_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/clusterBed/clusterMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMaxDistance\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveMaxDistance\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/clusterBed/clusterBed.cpp\n",
            "  * compiling src/complementFile/complementHelp.cpp\n",
            "  * compiling src/complementFile/complementFile.cpp\n",
            "  * compiling src/coverageFile/coverageHelp.cpp\n",
            "  * compiling src/coverageFile/coverageFile.cpp\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvirtual void CoverageFile::giveFinalReport(RecordOutputMgr*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:100:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  100 |                 \u001b[01;35m\u001b[Kasprintf(&depthPctString, \"%0.7f\", depthPct)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doMean(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:181:17:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  181 |         \u001b[01;35m\u001b[Kasprintf(&meanString, \"%0.7f\", mean)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doHist(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:210:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  210 |                 \u001b[01;35m\u001b[Kasprintf(&coveredFractionString, \"%0.7f\", coveredFraction)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doDefault(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:230:17:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  230 |         \u001b[01;35m\u001b[Kasprintf(&coveredFractionString, \"%0.7f\", coveredFraction)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/expand/expand.cpp\n",
            "  * compiling src/fastaFromBed/fastaFromBedMain.cpp\n",
            "  * compiling src/fastaFromBed/fastaFromBed.cpp\n",
            "  * compiling src/flankBed/flankBedMain.cpp\n",
            "  * compiling src/flankBed/flankBed.cpp\n",
            "  * compiling src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "  * compiling src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/GenomeFile/NewGenomeFile.h:20\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/genomeCoverageBed/genomeCoverageBed.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/genomeCoverageBed/genomeCoverageBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/getOverlap/getOverlap.cpp\n",
            "\u001b[01m\u001b[Ksrc/getOverlap/getOverlap.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint getoverlap_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/getOverlap/getOverlap.cpp:44:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveColumns\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   44 |     bool \u001b[01;35m\u001b[KhaveColumns\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/groupBy/groupByHelp.cpp\n",
            "  * compiling src/groupBy/groupBy.cpp\n",
            "  * compiling src/intersectFile/intersectHelp.cpp\n",
            "  * compiling src/intersectFile/intersectFile.cpp\n",
            "  * compiling src/fisher/fisherHelp.cpp\n",
            "  * compiling src/fisher/fisher.cpp\n",
            "  * compiling src/fisher/kfunc.cpp\n",
            "  * compiling src/jaccard/jaccardHelp.cpp\n",
            "  * compiling src/jaccard/jaccard.cpp\n",
            "  * compiling src/linksBed/linksMain.cpp\n",
            "  * compiling src/linksBed/linksBed.cpp\n",
            "  * compiling src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "  * compiling src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "  * compiling src/mapFile/mapHelp.cpp\n",
            "  * compiling src/mapFile/mapFile.cpp\n",
            "  * compiling src/mergeFile/mergeHelp.cpp\n",
            "  * compiling src/mergeFile/mergeFile.cpp\n",
            "  * compiling src/multiBamCov/multiBamCovMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint multibamcov_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveBed\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveBed\u001b[m\u001b[K           = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:39:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveBams\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   39 |     bool \u001b[01;35m\u001b[KhaveBams\u001b[m\u001b[K          = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:47:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   47 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K       = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiBamCov/multiBamCov.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/BamReader.hpp:4\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamMultiReader.h:4\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/multiBamCov/multiBamCov.h:16\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/multiBamCov/multiBamCov.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint multiintersect_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:45:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiles\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   45 |     bool \u001b[01;35m\u001b[KhaveFiles\u001b[m\u001b[K         = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:47:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveGenome\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   47 |     bool \u001b[01;35m\u001b[KhaveGenome\u001b[m\u001b[K        = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:48:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiller\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   48 |     bool \u001b[01;35m\u001b[KhaveFiller\u001b[m\u001b[K        = true;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiIntersectBed/multiIntersectBed.cpp\n",
            "  * compiling src/nucBed/nucBedMain.cpp\n",
            "  * compiling src/nucBed/nucBed.cpp\n",
            "  * compiling src/pairToBed/pairToBedMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBedMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint pairtobed_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBedMain.cpp:43:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   43 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K       = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToBed/pairToBed.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid BedIntersectPE::FindSpanningOverlaps(const BEDPE&, std::vector<BED>&, const string&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:257:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KspanLength\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  257 |     CHRPOS \u001b[01;35m\u001b[KspanLength\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool BedIntersectPE::FindOneOrMoreSpanningOverlaps(const BEDPE&, const string&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:310:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KspanLength\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  310 |     CHRPOS \u001b[01;35m\u001b[KspanLength\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/pairToBed/pairToBed.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToPair/pairToPairMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPairMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint pairtopair_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPairMain.cpp:44:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   44 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToPair/pairToPair.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid PairToPair::FindOverlaps(const BEDPE&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:112:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kfound1\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  112 |         bool \u001b[01;35m\u001b[Kfound1\u001b[m\u001b[K = false;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:113:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kfound2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  113 |         bool \u001b[01;35m\u001b[Kfound2\u001b[m\u001b[K = false;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/randomBed/randomBedMain.cpp\n",
            "  * compiling src/randomBed/randomBed.cpp\n",
            "  * compiling src/regressTest/regressTestMain.cpp\n",
            "  * compiling src/regressTest/RegressTest.cpp\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::executeAndCompareCorrectness(const fileListType&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:431:15:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint system(const char*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  431 |         \u001b[01;35m\u001b[Ksystem(diffCmd.c_str())\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::startMemoryProfile(bool)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:573:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* fgets(char*, int, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  573 |         \u001b[01;35m\u001b[Kfgets(sLine, 4192, fp)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::calcMemoryStats()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:622:22:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* fgets(char*, int, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  622 |                 \u001b[01;35m\u001b[Kfgets(sLine, 4192, fp)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/reldist/reldistMain.cpp\n",
            "  * compiling src/reldist/reldist.cpp\n",
            "\u001b[01m\u001b[Ksrc/reldist/reldist.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RelativeDistance::CalculateRelativeDistance()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/reldist/reldist.cpp:129:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[KCHRPOS\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong int\u001b[m\u001b[K’} and ‘\u001b[01m\u001b[Kstd::vector<long int>::size_type\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  129 |             if (\u001b[01;35m\u001b[Klow_idx != chrom_mids->size() - 1\u001b[m\u001b[K)\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/sampleFile/sampleHelp.cpp\n",
            "  * compiling src/sampleFile/sampleFile.cpp\n",
            "  * compiling src/shiftBed/shiftBedMain.cpp\n",
            "  * compiling src/shiftBed/shiftBed.cpp\n",
            "  * compiling src/shuffleBed/shuffleBedMain.cpp\n",
            "  * compiling src/shuffleBed/shuffleBed.cpp\n",
            "  * compiling src/slopBed/slopBedMain.cpp\n",
            "  * compiling src/slopBed/slopBed.cpp\n",
            "  * compiling src/sortBed/sortMain.cpp\n",
            "  * compiling src/sortBed/sortBed.cpp\n",
            "  * compiling src/spacingFile/spacingHelp.cpp\n",
            "  * compiling src/spacingFile/spacingFile.cpp\n",
            "  * compiling src/split/splitBedMain.cpp\n",
            "  * compiling src/split/splitBed.cpp\n",
            "  * compiling src/subtractFile/subtractHelp.cpp\n",
            "  * compiling src/subtractFile/subtractFile.cpp\n",
            "  * compiling src/summaryFile/summaryHelp.cpp\n",
            "  * compiling src/summaryFile/summaryFile.cpp\n",
            "  * compiling src/tagBam/tagBamMain.cpp\n",
            "  * compiling src/tagBam/tagBam.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/tagBam/tagBam.h:17\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/tagBam/tagBam.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint unionbedgraphs_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:46:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiles\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   46 |     bool \u001b[01;35m\u001b[KhaveFiles\u001b[m\u001b[K         = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:48:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveGenome\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   48 |     bool \u001b[01;35m\u001b[KhaveGenome\u001b[m\u001b[K        = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:49:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiller\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   49 |     bool \u001b[01;35m\u001b[KhaveFiller\u001b[m\u001b[K        = true;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/unionBedGraphs/unionBedGraphs.cpp\n",
            "  * compiling src/windowBed/windowMain.cpp\n",
            "  * compiling src/windowBed/windowBed.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/windowBed/windowBed.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/windowBed/windowBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/windowMaker/windowMakerMain.cpp\n",
            "  * compiling src/windowMaker/windowMaker.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool BufferedStreamMgr::readFileChunk()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp:148:37:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  148 |                 if (\u001b[01;35m\u001b[K_mainBufCurrLen < _useBufSize\u001b[m\u001b[K) {\n",
            "      |                     \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~^~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/InputStreamMgr.h:14\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp:8\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:30:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |                 \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/Records/Record.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/Records/BlockMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kint BlockMgr::getNonRedundantOverlap()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/Records/BlockMgr.cpp:194:35:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Kstd::vector<long int>::size_type\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  194 |                 for (int i = 1; \u001b[01;35m\u001b[Ki < _overlapStarts.size()\u001b[m\u001b[K; ++i)\n",
            "      |                                 \u001b[01;35m\u001b[K~~^~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "  * compiling src/utils/bedFile/bedFile.cpp\n",
            "  * compiling src/utils/BinTree/BinTree.cpp\n",
            "  * compiling src/utils/version/version.cpp\n",
            "  * compiling src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "  * compiling src/utils/chromsweep/chromsweep.cpp\n",
            "  * compiling src/utils/Contexts/ContextBase.cpp\n",
            "  * compiling src/utils/Contexts/ContextIntersect.cpp\n",
            "  * compiling src/utils/Contexts/ContextFisher.cpp\n",
            "  * compiling src/utils/Contexts/ContextMap.cpp\n",
            "  * compiling src/utils/Contexts/ContextSample.cpp\n",
            "  * compiling src/utils/Contexts/ContextSpacing.cpp\n",
            "  * compiling src/utils/Contexts/ContextMerge.cpp\n",
            "  * compiling src/utils/Contexts/ContextJaccard.cpp\n",
            "  * compiling src/utils/Contexts/ContextClosest.cpp\n",
            "  * compiling src/utils/Contexts/ContextSubtract.cpp\n",
            "  * compiling src/utils/Contexts/ContextCoverage.cpp\n",
            "  * compiling src/utils/Contexts/ContextComplement.cpp\n",
            "  * compiling src/utils/Contexts/ContextGroupBy.cpp\n",
            "  * compiling src/utils/Contexts/ContextSummary.cpp\n",
            "  * compiling src/utils/general/ParseTools.cpp\n",
            "  * compiling src/utils/general/PushBackStreamBuf.cpp\n",
            "  * compiling src/utils/general/CompressionTools.cpp\n",
            "  * compiling src/utils/general/Tokenizer.cpp\n",
            "  * compiling src/utils/general/CommonHelp.cpp\n",
            "  * compiling src/utils/general/ErrorMsg.cpp\n",
            "  * compiling src/utils/general/Random.cpp\n",
            "g++ -g -Wall -O2 -std=c++11  -Isrc/utils/gzstream  -c -o obj/gzstream.o src/utils/gzstream/gzstream.C\n",
            "  * compiling src/utils/fileType/fileType.cpp\n",
            "  * compiling src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/fileType/FileRecordTypeChecker.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool FileRecordTypeChecker::isTextDelimtedFormat(const char*, size_t)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/fileType/FileRecordTypeChecker.cpp:366:33:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  366 |                 for (int j=0; \u001b[01;35m\u001b[Kj < len\u001b[m\u001b[K; j++) {\n",
            "      |                               \u001b[01;35m\u001b[K~~^~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/bedFilePE/bedFilePE.cpp\n",
            "  * compiling src/utils/KeyListOps/KeyListOps.cpp\n",
            "  * compiling src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "  * compiling src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "  * compiling src/utils/NewChromsweep/CloseSweep.cpp\n",
            "  * compiling src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "  * compiling src/utils/tabFile/tabFile.cpp\n",
            "  * compiling src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "  * compiling src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "  * compiling src/utils/Fasta/Fasta.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kstd::string FastaReference::getSequence(std::string)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:292:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Ksize_t fread(void*, size_t, size_t, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  292 |         \u001b[01;35m\u001b[Kfread(seq, sizeof(char), seqlen, file)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kstd::string FastaReference::getSubSequence(std::string, CHRPOS, CHRPOS)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:334:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Ksize_t fread(void*, size_t, size_t, FILE*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  334 |         \u001b[01;35m\u001b[Kfread(seq, sizeof(char), (off_t) seqlen, file)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/Fasta/split.cpp\n",
            "  * compiling src/utils/VectorOps/VectorOps.cpp\n",
            "  * compiling src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "  * compiling src/utils/GenomeFile/GenomeFile.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/GenomeFile/GenomeFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid GenomeFile::loadGenomeFileIntoMap()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/GenomeFile/GenomeFile.cpp:60:26:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kc2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   60 |                     long \u001b[01;35m\u001b[Kc2\u001b[m\u001b[K;\n",
            "      |                          \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "  * compiling src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RecordOutputMgr::printClosest(RecordKeyVector&, const std::vector<long int>*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:170:56:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  170 |                 if (\u001b[01;35m\u001b[Kcontext->getNumClosestHitsWanted() > keyList.size()\u001b[m\u001b[K)\n",
            "      |                     \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RecordOutputMgr::reportOverlapSummary(RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:455:38:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} and ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare\u0007-Wsign-compare\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  455 |                 for (size_t i = 1; \u001b[01;35m\u001b[Ki < _context->getNumInputFiles()\u001b[m\u001b[K; i++)\n",
            "      |                                    \u001b[01;35m\u001b[K~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/ToolBase/ToolBase.cpp\n",
            "  * compiling src/utils/driver/BedtoolsDriver.cpp\n",
            "cd src/utils/htslib && make config.h\n",
            "make[1]: Entering directory '/content/bedtools2/src/utils/htslib'\n",
            "echo '/* Default config.h generated by Makefile */' > config.h\n",
            "echo '#define HAVE_LIBBZ2 1' >> config.h\n",
            "echo '#define HAVE_LIBLZMA 1' >> config.h\n",
            "echo '#ifndef __APPLE__' >> config.h\n",
            "echo '#define HAVE_LZMA_H 1' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#define HAVE_FSEEKO 1' >> config.h\n",
            "echo '#define HAVE_DRAND48 1' >> config.h\n",
            "make[1]: Leaving directory '/content/bedtools2/src/utils/htslib'\n",
            "cd src/utils/htslib && make lib-static\n",
            "make[1]: Entering directory '/content/bedtools2/src/utils/htslib'\n",
            "gcc -g -Wall -O2 -I.  -c -o kfunc.o kfunc.c\n",
            "gcc -g -Wall -O2 -I.  -c -o knetfile.o knetfile.c\n",
            "gcc -g -Wall -O2 -I.  -c -o kstring.o kstring.c\n",
            "gcc -g -Wall -O2 -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c\n",
            "gcc -g -Wall -O2 -I.  -c -o bgzf.o bgzf.c\n",
            "gcc -g -Wall -O2 -I.  -c -o errmod.o errmod.c\n",
            "gcc -g -Wall -O2 -I.  -c -o faidx.o faidx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hfile.o hfile.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hfile_net.o hfile_net.c\n",
            "echo '#define HTS_VERSION \"1.9\"' > version.h\n",
            "gcc -g -Wall -O2 -I.  -c -o hts.o hts.c\n",
            "\u001b[01m\u001b[Khts.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Khts_idx_init\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Khts.c:1397:59:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Koverflow in conversion from ‘\u001b[01m\u001b[Kuint32_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kunsigned int\u001b[m\u001b[K’} to ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ changes value from ‘\u001b[01m\u001b[Kidx->z.last_bin = 4294967295\u001b[m\u001b[K’ to ‘\u001b[01m\u001b[K-1\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverflow\u0007-Woverflow\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1397 |     idx->z.save_bin = idx->z.save_tid = idx->z.last_tid = \u001b[01;35m\u001b[Kidx\u001b[m\u001b[K->z.last_bin = 0xffffffffu;\n",
            "      |                                                           \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n",
            "gcc -g -Wall -O2 -I.  -c -o hts_os.o hts_os.c\n",
            "gcc -g -Wall -O2 -I.  -c -o md5.o md5.c\n",
            "gcc -g -Wall -O2 -I.  -c -o multipart.o multipart.c\n",
            "gcc -g -Wall -O2 -I.  -c -o probaln.o probaln.c\n",
            "gcc -g -Wall -O2 -I.  -c -o realn.o realn.c\n",
            "gcc -g -Wall -O2 -I.  -c -o regidx.o regidx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o sam.o sam.c\n",
            "gcc -g -Wall -O2 -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcf_sweep.o vcf_sweep.c\n",
            "gcc -g -Wall -O2 -I.  -c -o tbx.o tbx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o textutils.o textutils.c\n",
            "gcc -g -Wall -O2 -I.  -c -o thread_pool.o thread_pool.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcf.o vcf.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcfutils.o vcfutils.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_decode.o cram/cram_decode.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_encode.o cram/cram_encode.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_external.o cram/cram_external.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_index.o cram/cram_index.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_io.o cram/cram_io.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/cram_io.c:53\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_populate_ref\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2095:9\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3748:13\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3823:17\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_dopen\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:4126:9\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 20 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_samtools.o cram/cram_samtools.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_stats.o cram/cram_stats.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/files.o cram/files.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/mFILE.o cram/mFILE.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/rANS_static.o cram/rANS_static.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/sam_header.o cram/sam_header.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/string_alloc.o cram/string_alloc.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/string_alloc.c:41\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_ndup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:151:5\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:143:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying as many bytes from a string as its length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:143:12:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Klength computed here\n",
            "  143 |     return \u001b[01;36m\u001b[Kstring_ndup(a_str, instr, strlen(instr))\u001b[m\u001b[K;\n",
            "      |            \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "ar -rc libhts.a kfunc.o knetfile.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o hfile.o hfile_net.o hts.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o\n",
            "ranlib libhts.a\n",
            "make[1]: Leaving directory '/content/bedtools2/src/utils/htslib'\n",
            "- Building main bedtools binary.\n",
            "done.\n",
            "- Creating executables for old CLI.\n",
            "done.\n",
            "  * compiling src/htsutil.cpp\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp /content/bedtools2/bin/* /usr/local/bin/"
      ],
      "metadata": {
        "id": "h3fz1s2UGiEv"
      },
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! wget https://github.com/bedops/bedops/releases/download/v2.4.41/bedops_linux_x86_64-v2.4.41.tar.bz2\n",
        "! tar jxvf bedops_linux_x86_64-v2.4.41.tar.bz2\n",
        "! cp bin/* /usr/local/bin"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IW4mQsspKhii",
        "outputId": "7746850f-1415-47ec-b137-1407ba2c269f"
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2024-06-16 12:45:35--  https://github.com/bedops/bedops/releases/download/v2.4.41/bedops_linux_x86_64-v2.4.41.tar.bz2\n",
            "Resolving github.com (github.com)... 20.205.243.166\n",
            "Connecting to github.com (github.com)|20.205.243.166|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/12932856/7baae005-767f-4700-bd69-68f44f9a01bf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240616%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240616T124535Z&X-Amz-Expires=300&X-Amz-Signature=bbc0793aec22ffe9cfe305fd00cebc9fc63d417ca5cac0ad9097aa1ae3d551a9&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=12932856&response-content-disposition=attachment%3B%20filename%3Dbedops_linux_x86_64-v2.4.41.tar.bz2&response-content-type=application%2Foctet-stream [following]\n",
            "--2024-06-16 12:45:35--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/12932856/7baae005-767f-4700-bd69-68f44f9a01bf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240616%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240616T124535Z&X-Amz-Expires=300&X-Amz-Signature=bbc0793aec22ffe9cfe305fd00cebc9fc63d417ca5cac0ad9097aa1ae3d551a9&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=12932856&response-content-disposition=attachment%3B%20filename%3Dbedops_linux_x86_64-v2.4.41.tar.bz2&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 26814112 (26M) [application/octet-stream]\n",
            "Saving to: ‘bedops_linux_x86_64-v2.4.41.tar.bz2’\n",
            "\n",
            "bedops_linux_x86_64 100%[===================>]  25.57M  --.-KB/s    in 0.07s   \n",
            "\n",
            "2024-06-16 12:45:36 (380 MB/s) - ‘bedops_linux_x86_64-v2.4.41.tar.bz2’ saved [26814112/26814112]\n",
            "\n",
            "bin/\n",
            "bin/bam2bed-typical\n",
            "bin/bam2bed-megarow\n",
            "bin/bam2bed-float128\n",
            "bin/bam2bed_gnuParallel-typical\n",
            "bin/bam2bed_gnuParallel-megarow\n",
            "bin/bam2bed_gnuParallel-float128\n",
            "bin/bam2bed_sge-typical\n",
            "bin/bam2bed_sge-megarow\n",
            "bin/bam2bed_sge-float128\n",
            "bin/bam2bed_slurm-typical\n",
            "bin/bam2bed_slurm-megarow\n",
            "bin/bam2bed_slurm-float128\n",
            "bin/bam2starch-typical\n",
            "bin/bam2starch-megarow\n",
            "bin/bam2starch-float128\n",
            "bin/bam2starch_gnuParallel-typical\n",
            "bin/bam2starch_gnuParallel-megarow\n",
            "bin/bam2starch_gnuParallel-float128\n",
            "bin/bam2starch_sge-typical\n",
            "bin/bam2starch_sge-megarow\n",
            "bin/bam2starch_sge-float128\n",
            "bin/bam2starch_slurm-typical\n",
            "bin/bam2starch_slurm-megarow\n",
            "bin/bam2starch_slurm-float128\n",
            "bin/gff2bed-typical\n",
            "bin/gff2bed-megarow\n",
            "bin/gff2bed-float128\n",
            "bin/gff2starch-typical\n",
            "bin/gff2starch-megarow\n",
            "bin/gff2starch-float128\n",
            "bin/gtf2bed-typical\n",
            "bin/gtf2bed-megarow\n",
            "bin/gtf2bed-float128\n",
            "bin/gtf2starch-typical\n",
            "bin/gtf2starch-megarow\n",
            "bin/gtf2starch-float128\n",
            "bin/gvf2bed-typical\n",
            "bin/gvf2bed-megarow\n",
            "bin/gvf2bed-float128\n",
            "bin/gvf2starch-typical\n",
            "bin/gvf2starch-megarow\n",
            "bin/gvf2starch-float128\n",
            "bin/psl2bed-typical\n",
            "bin/psl2bed-megarow\n",
            "bin/psl2bed-float128\n",
            "bin/psl2starch-typical\n",
            "bin/psl2starch-megarow\n",
            "bin/psl2starch-float128\n",
            "bin/rmsk2bed-typical\n",
            "bin/rmsk2bed-megarow\n",
            "bin/rmsk2bed-float128\n",
            "bin/rmsk2starch-typical\n",
            "bin/rmsk2starch-megarow\n",
            "bin/rmsk2starch-float128\n",
            "bin/sam2bed-typical\n",
            "bin/sam2bed-megarow\n",
            "bin/sam2bed-float128\n",
            "bin/sam2starch-typical\n",
            "bin/sam2starch-megarow\n",
            "bin/sam2starch-float128\n",
            "bin/vcf2bed-typical\n",
            "bin/vcf2bed-megarow\n",
            "bin/vcf2bed-float128\n",
            "bin/vcf2starch-typical\n",
            "bin/vcf2starch-megarow\n",
            "bin/vcf2starch-float128\n",
            "bin/wig2bed-typical\n",
            "bin/wig2bed-megarow\n",
            "bin/wig2bed-float128\n",
            "bin/wig2starch-typical\n",
            "bin/wig2starch-megarow\n",
            "bin/wig2starch-float128\n",
            "bin/starchcluster_sge-typical\n",
            "bin/starchcluster_gnuParallel-typical\n",
            "bin/starchcluster_slurm-typical\n",
            "bin/starch-diff-typical\n",
            "bin/starchcluster_sge-megarow\n",
            "bin/starchcluster_gnuParallel-megarow\n",
            "bin/starchcluster_slurm-megarow\n",
            "bin/starch-diff-megarow\n",
            "bin/starchcluster_sge-float128\n",
            "bin/starchcluster_gnuParallel-float128\n",
            "bin/starchcluster_slurm-float128\n",
            "bin/starch-diff-float128\n",
            "bin/sort-bed-typical\n",
            "bin/update-sort-bed-slurm-typical\n",
            "bin/update-sort-bed-starch-slurm-typical\n",
            "bin/update-sort-bed-migrate-candidates-typical\n",
            "bin/bedops-typical\n",
            "bin/closest-features-typical\n",
            "bin/bedmap-typical\n",
            "bin/bedextract-typical\n",
            "bin/starch-typical\n",
            "bin/unstarch-typical\n",
            "bin/starchcat-typical\n",
            "bin/starchstrip-typical\n",
            "bin/convert2bed-typical\n",
            "bin/sort-bed-megarow\n",
            "bin/update-sort-bed-slurm-megarow\n",
            "bin/update-sort-bed-starch-slurm-megarow\n",
            "bin/update-sort-bed-migrate-candidates-megarow\n",
            "bin/bedops-megarow\n",
            "bin/closest-features-megarow\n",
            "bin/bedmap-megarow\n",
            "bin/bedextract-megarow\n",
            "bin/starch-megarow\n",
            "bin/unstarch-megarow\n",
            "bin/starchcat-megarow\n",
            "bin/starchstrip-megarow\n",
            "bin/convert2bed-megarow\n",
            "bin/sort-bed-float128\n",
            "bin/update-sort-bed-slurm-float128\n",
            "bin/update-sort-bed-starch-slurm-float128\n",
            "bin/update-sort-bed-migrate-candidates-float128\n",
            "bin/bedops-float128\n",
            "bin/closest-features-float128\n",
            "bin/bedmap-float128\n",
            "bin/bedextract-float128\n",
            "bin/starch-float128\n",
            "bin/unstarch-float128\n",
            "bin/starchcat-float128\n",
            "bin/starchstrip-float128\n",
            "bin/convert2bed-float128\n",
            "bin/switch-BEDOPS-binary-type\n",
            "bin/bam2bed\n",
            "bin/bam2bed_gnuParallel\n",
            "bin/bam2bed_sge\n",
            "bin/bam2bed_slurm\n",
            "bin/bam2starch\n",
            "bin/bam2starch_gnuParallel\n",
            "bin/bam2starch_sge\n",
            "bin/bam2starch_slurm\n",
            "bin/gff2bed\n",
            "bin/gff2starch\n",
            "bin/gtf2bed\n",
            "bin/gtf2starch\n",
            "bin/gvf2bed\n",
            "bin/gvf2starch\n",
            "bin/psl2bed\n",
            "bin/psl2starch\n",
            "bin/rmsk2bed\n",
            "bin/rmsk2starch\n",
            "bin/sam2bed\n",
            "bin/sam2starch\n",
            "bin/vcf2bed\n",
            "bin/vcf2starch\n",
            "bin/wig2bed\n",
            "bin/wig2starch\n",
            "bin/starchcluster_sge\n",
            "bin/starchcluster_gnuParallel\n",
            "bin/starchcluster_slurm\n",
            "bin/starch-diff\n",
            "bin/sort-bed\n",
            "bin/update-sort-bed-slurm\n",
            "bin/update-sort-bed-starch-slurm\n",
            "bin/update-sort-bed-migrate-candidates\n",
            "bin/bedops\n",
            "bin/closest-features\n",
            "bin/bedmap\n",
            "bin/bedextract\n",
            "bin/starch\n",
            "bin/unstarch\n",
            "bin/starchcat\n",
            "bin/starchstrip\n",
            "bin/convert2bed\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "С .gff неудобно работать, потому переделаем его в .bed"
      ],
      "metadata": {
        "id": "zOBbWBvRdg4I"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sortBed -i genomic.gff | gff2bed --do-not-sort > genomic.bed"
      ],
      "metadata": {
        "id": "U6rvw2KlG5bz",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2576a6ee-1a10-4dda-c0f4-8dc0811f50ae"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Error: The requested file (genomic.gff) could not be opened. Error message: (No such file or directory). Exiting!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Тоже самое сделаем с выводом нейросети"
      ],
      "metadata": {
        "id": "i9giYda6d4sn"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import re\n",
        "\n",
        "gene = \"\"\n",
        "with open (\"text_predictions.txt\", \"r\") as f:\n",
        "  with open(\"bed_predictions1.bed\", \"w\") as o:\n",
        "    while s := f.readline():\n",
        "      if \"NW\" in s:\n",
        "        gene = s.rstrip()\n",
        "      elif (\"GCF\" in s) or (\"start\" in s):\n",
        "        pass\n",
        "      else:\n",
        "        mas = re.findall(r'\\d+', s)\n",
        "        o.write(f\"{gene}\\t{mas[0]}\\t{mas[1]}\\n\")"
      ],
      "metadata": {
        "id": "y5HT1FrNM1VK"
      },
      "execution_count": 15,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Теперь можно прочесть файл"
      ],
      "metadata": {
        "id": "wrJjF2wGeKMN"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "\n",
        "genome = pd.read_csv(\"genomic.bed\", names=[\"chrom\", \"start\", \"end\", \"score1\", \"score2\", \"strand\", \"lab\", \"type\", \"phase\", \"attributes\"], sep=\"\\t\")\n",
        "\n",
        "genome"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 53
        },
        "id": "kAOhu8T1r9GN",
        "outputId": "ac4361cf-252d-409c-8ba3-6e23855626d7"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "Empty DataFrame\n",
              "Columns: [chrom, start, end, score1, score2, strand, lab, type, phase, attributes]\n",
              "Index: []"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-cf654a9e-330e-4dad-8cca-40b2ae874366\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chrom</th>\n",
              "      <th>start</th>\n",
              "      <th>end</th>\n",
              "      <th>score1</th>\n",
              "      <th>score2</th>\n",
              "      <th>strand</th>\n",
              "      <th>lab</th>\n",
              "      <th>type</th>\n",
              "      <th>phase</th>\n",
              "      <th>attributes</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-cf654a9e-330e-4dad-8cca-40b2ae874366')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-cf654a9e-330e-4dad-8cca-40b2ae874366 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-cf654a9e-330e-4dad-8cca-40b2ae874366');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "genome",
              "repr_error": "Out of range float values are not JSON compliant: nan"
            }
          },
          "metadata": {},
          "execution_count": 25
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Квадруплексы надо в пандас для удобства"
      ],
      "metadata": {
        "id": "RQpmIqVxeZHM"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pqs = pd.read_csv(\"pqs.bed\", names=[\"chrom\", \"start\", \"end\"], sep=\"\\t\")\n",
        "\n",
        "pqs"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 423
        },
        "id": "SAD8Xa7ju-O0",
        "outputId": "852c1cab-eb2a-4a19-9eff-2b7191c4faa2"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                chrom   start     end\n",
              "0   CAJJDO010000002.1  534196  534314\n",
              "1   CAJJDO010000009.1  519367  519397\n",
              "2   CAJJDO010000010.1   67153   67174\n",
              "3   CAJJDO010000012.1  289822  289851\n",
              "4   CAJJDO010000012.1       0      76\n",
              "..                ...     ...     ...\n",
              "86  CAJJDO010000222.1    3603    3643\n",
              "87  CAJJDO010000229.1       0      21\n",
              "88  CAJJDO010000242.1    8289    8376\n",
              "89  CAJJDO010000254.1   10069   10096\n",
              "90  CAJJDO010000259.1    7681    7757\n",
              "\n",
              "[91 rows x 3 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-be75b171-4692-47d0-812c-af907029d5fd\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chrom</th>\n",
              "      <th>start</th>\n",
              "      <th>end</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>CAJJDO010000002.1</td>\n",
              "      <td>534196</td>\n",
              "      <td>534314</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>CAJJDO010000009.1</td>\n",
              "      <td>519367</td>\n",
              "      <td>519397</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>CAJJDO010000010.1</td>\n",
              "      <td>67153</td>\n",
              "      <td>67174</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>CAJJDO010000012.1</td>\n",
              "      <td>289822</td>\n",
              "      <td>289851</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>CAJJDO010000012.1</td>\n",
              "      <td>0</td>\n",
              "      <td>76</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>86</th>\n",
              "      <td>CAJJDO010000222.1</td>\n",
              "      <td>3603</td>\n",
              "      <td>3643</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>87</th>\n",
              "      <td>CAJJDO010000229.1</td>\n",
              "      <td>0</td>\n",
              "      <td>21</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>88</th>\n",
              "      <td>CAJJDO010000242.1</td>\n",
              "      <td>8289</td>\n",
              "      <td>8376</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>89</th>\n",
              "      <td>CAJJDO010000254.1</td>\n",
              "      <td>10069</td>\n",
              "      <td>10096</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>90</th>\n",
              "      <td>CAJJDO010000259.1</td>\n",
              "      <td>7681</td>\n",
              "      <td>7757</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>91 rows × 3 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-be75b171-4692-47d0-812c-af907029d5fd')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-be75b171-4692-47d0-812c-af907029d5fd button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-be75b171-4692-47d0-812c-af907029d5fd');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-e13e9f63-5003-43bb-a1c1-e38f16275c50\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-e13e9f63-5003-43bb-a1c1-e38f16275c50')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-e13e9f63-5003-43bb-a1c1-e38f16275c50 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "pqs",
              "summary": "{\n  \"name\": \"pqs\",\n  \"rows\": 91,\n  \"fields\": [\n    {\n      \"column\": \"chrom\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 67,\n        \"samples\": [\n          \"CAJJDO010000106.1\",\n          \"CAJJDO010000045.1\",\n          \"CAJJDO010000016.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"start\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 204527,\n        \"min\": 0,\n        \"max\": 755057,\n        \"num_unique_values\": 83,\n        \"samples\": [\n          300485,\n          534196,\n          102321\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"end\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 204527,\n        \"min\": 21,\n        \"max\": 755083,\n        \"num_unique_values\": 89,\n        \"samples\": [\n          43,\n          12875,\n          379275\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 27
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Вспомогательная функция для поиска межгенных попаданий"
      ],
      "metadata": {
        "id": "lPI7LfbBeeS3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_between(gen, needle, find_type: str):\n",
        "  ex1 = None\n",
        "  ex2 = None\n",
        "  ned_types = gen.loc[((gen[\"type\"] == find_type) & (gen[\"chrom\"] == needle[\"chrom\"]))]\n",
        "  for i in range(len(ned_types) - 1):\n",
        "      if ex1 is None:\n",
        "        ex1 = ned_types.iloc[[i]]\n",
        "      else:\n",
        "        ex2 = ned_types.iloc[[i]]\n",
        "        if ((ex1[\"end\"].item() <= needle[\"start\"]) and (ex2[\"start\"].item() >= needle[\"end\"])) \\\n",
        "          or ((ex1[\"start\"].item() <= needle[\"start\"]) and (ex2[\"start\"].item() >= needle[\"end\"])) \\\n",
        "          or ((ex1[\"end\"].item() <= needle[\"start\"]) and (ex2[\"end\"].item() >= needle[\"end\"])) \\\n",
        "          or ((ex1[\"start\"].item() <= needle[\"start\"]) and (ex2[\"end\"].item() >= needle[\"end\"])\n",
        "          ):\n",
        "            return [ex1, ex2]\n",
        "\n",
        "\n",
        "        ex1 = ex2\n",
        "  else:\n",
        "    return []"
      ],
      "metadata": {
        "id": "HYJ5iKIvPOHH"
      },
      "execution_count": 28,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Большая функция для поиска вхождений всех структур, а также возвращения генов в промотеры которых попали они"
      ],
      "metadata": {
        "id": "LbNdFZWMemXC"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_count(genome, pqs):\n",
        "  searching_genes = {}\n",
        "  exons_count = 0\n",
        "  psq_count = 0\n",
        "  down_count = 0\n",
        "  intron_count = 0\n",
        "  intergenic_count = 0\n",
        "\n",
        "  for i in pqs.iterrows():\n",
        "    print(i[1])\n",
        "    exon_pattern = genome.loc[(\n",
        "      (genome[\"type\"] == \"exon\") & (genome[\"chrom\"] == i[1][\"chrom\"]) &\n",
        "      (\n",
        "          ((genome[\"start\"] <= i[1][\"start\"]) & (genome[\"end\"] >= i[1][\"end\"]))\n",
        "            | ((genome[\"start\"] <= i[1][\"end\"]) & (genome[\"start\"] >= i[1][\"start\"]))\n",
        "            | ((genome[\"end\"] <= i[1][\"end\"]) & (genome[\"start\"] <= i[1][\"start\"]) & (genome[\"end\"] >= i[1][\"start\"]))\n",
        "            | ((genome[\"start\"] >= i[1][\"start\"]) & (genome[\"end\"] <= i[1][\"end\"]))\n",
        "        )\n",
        "    )]\n",
        "\n",
        "    psq_pattern_plus = genome.loc[(\n",
        "      (genome[\"strand\"] == \"+\") & (genome[\"type\"] == \"gene\") & (genome[\"chrom\"] == i[1][\"chrom\"]) &\n",
        "      (\n",
        "          ((genome[\"start\"] < i[1][\"start\"]) & (genome[\"end\"] > i[1][\"end\"]))\n",
        "              | (((genome[\"start\"] - 1000) <= i[1][\"end\"]) & ((genome[\"start\"] - 1000) >= i[1][\"start\"]))\n",
        "              | (((genome[\"start\"] - 1000) <= i[1][\"start\"]) & (genome[\"start\"] >= i[1][\"start\"]))\n",
        "      )\n",
        "    )]\n",
        "\n",
        "    psq_pattern_minus = genome.loc[(\n",
        "      (genome[\"strand\"] == \"-\") & (genome[\"type\"] == \"gene\") & (genome[\"chrom\"] == i[1][\"chrom\"]) &\n",
        "      (\n",
        "          ((genome[\"start\"] < i[1][\"start\"]) & (genome[\"end\"] > i[1][\"end\"]))\n",
        "              | (((genome[\"end\"] + 1000) <= i[1][\"end\"]) & ((genome[\"end\"] + 1000) >= i[1][\"start\"]))\n",
        "              | (((genome[\"end\"] + 1000) >= i[1][\"end\"]) & (genome[\"end\"] <= i[1][\"end\"]))\n",
        "      )\n",
        "    )]\n",
        "\n",
        "\n",
        "    down_pattern_plus = genome.loc[(\n",
        "      (genome[\"strand\"] == \"+\") & (genome[\"type\"] == \"gene\") & (genome[\"chrom\"] == i[1][\"chrom\"]) &\n",
        "      (\n",
        "          ((genome[\"start\"] < i[1][\"start\"]) & (genome[\"end\"] > i[1][\"end\"]))\n",
        "              | (((genome[\"end\"] + 200) <= i[1][\"end\"]) & ((genome[\"end\"] + 200) >= i[1][\"start\"]))\n",
        "              | (((genome[\"end\"] + 200) >= i[1][\"end\"]) & (genome[\"end\"] <= i[1][\"end\"]))\n",
        "      )\n",
        "    )]\n",
        "\n",
        "    down_pattern_minus = genome.loc[(\n",
        "      (genome[\"strand\"] == \"-\") & (genome[\"type\"] == \"gene\") & (genome[\"chrom\"] == i[1][\"chrom\"]) &\n",
        "      (\n",
        "          ((genome[\"start\"] < i[1][\"start\"]) & (genome[\"end\"] > i[1][\"end\"]))\n",
        "              | (((genome[\"start\"] - 200) <= i[1][\"end\"]) & ((genome[\"start\"] - 200) >= i[1][\"start\"]))\n",
        "              | (((genome[\"start\"] - 200) <= i[1][\"start\"]) & (genome[\"start\"] >= i[1][\"start\"]))\n",
        "      )\n",
        "    )]\n",
        "\n",
        "    intron_pattern = find_between(genome, i[1], \"exon\")\n",
        "    intergenic_pattern = find_between(genome, i[1], \"gene\")\n",
        "\n",
        "\n",
        "\n",
        "    if len(exon_pattern):\n",
        "      exons_count += 1\n",
        "    if len(psq_pattern_plus):\n",
        "      psq_count += 1\n",
        "      gene = psq_pattern_plus.iloc[[0]][\"attributes\"].item().split(\";\")[0]\n",
        "      if not searching_genes.get(gene):\n",
        "        searching_genes.update({gene: 1})\n",
        "    if len(psq_pattern_minus):\n",
        "      psq_count += 1\n",
        "      gene = psq_pattern_minus.iloc[[0]][\"attributes\"].item().split(\";\")[0]\n",
        "      if not searching_genes.get(gene):\n",
        "        searching_genes.update({gene: 1})\n",
        "    if len(down_pattern_plus):\n",
        "      down_count += 1\n",
        "    if len(down_pattern_minus):\n",
        "      down_count += 1\n",
        "    if len(intron_pattern):\n",
        "      intron_count += 1\n",
        "    if len(intergenic_pattern):\n",
        "      intergenic_count += 1\n",
        "\n",
        "\n",
        "\n",
        "  print(exons_count, psq_count, down_count, intron_count, intergenic_count)\n",
        "\n",
        "  return [exons_count, psq_count, down_count, intron_count, intergenic_count], searching_genes"
      ],
      "metadata": {
        "id": "DmpBxr3Lv65p"
      },
      "execution_count": 29,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Вызов для данных нейросети"
      ],
      "metadata": {
        "id": "gsDHvV6Qe1KT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd"
      ],
      "metadata": {
        "id": "iRQn6AX8zq-H"
      },
      "execution_count": 30,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Результаты"
      ],
      "metadata": {
        "id": "W48SxjKge9uT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "zdna = pd.read_csv(\"bed_predictions1.bed\", names=[\"chrom\", \"start\", \"end\"], sep=\"\\t\")\n",
        "\n",
        "zdna_counts, zdna_genes = find_count(genome, zdna)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JaphzNRoyEzR",
        "outputId": "38f47626-94ab-473f-d632-a5048838651a"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chrom            NaN\n",
            "start    905183005.0\n",
            "end              1.0\n",
            "Name: 0, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000001.0\n",
            "end             1.0\n",
            "Name: 1, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000002.0\n",
            "end             1.0\n",
            "Name: 2, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000003.0\n",
            "end             1.0\n",
            "Name: 3, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000004.0\n",
            "end             1.0\n",
            "Name: 4, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000005.0\n",
            "end             1.0\n",
            "Name: 5, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000006.0\n",
            "end             1.0\n",
            "Name: 6, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000007.0\n",
            "end             1.0\n",
            "Name: 7, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000008.0\n",
            "end             1.0\n",
            "Name: 8, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000009.0\n",
            "end             1.0\n",
            "Name: 9, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000010.0\n",
            "end             1.0\n",
            "Name: 10, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000011.0\n",
            "end             1.0\n",
            "Name: 11, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000012.0\n",
            "end             1.0\n",
            "Name: 12, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000013.0\n",
            "end             1.0\n",
            "Name: 13, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000014.0\n",
            "end             1.0\n",
            "Name: 14, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000015.0\n",
            "end             1.0\n",
            "Name: 15, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000016.0\n",
            "end             1.0\n",
            "Name: 16, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000017.0\n",
            "end             1.0\n",
            "Name: 17, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000018.0\n",
            "end             1.0\n",
            "Name: 18, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000019.0\n",
            "end             1.0\n",
            "Name: 19, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000020.0\n",
            "end             1.0\n",
            "Name: 20, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000021.0\n",
            "end             1.0\n",
            "Name: 21, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000022.0\n",
            "end             1.0\n",
            "Name: 22, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000023.0\n",
            "end             1.0\n",
            "Name: 23, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000024.0\n",
            "end             1.0\n",
            "Name: 24, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000025.0\n",
            "end             1.0\n",
            "Name: 25, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000026.0\n",
            "end             1.0\n",
            "Name: 26, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000027.0\n",
            "end             1.0\n",
            "Name: 27, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000028.0\n",
            "end             1.0\n",
            "Name: 28, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000029.0\n",
            "end             1.0\n",
            "Name: 29, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000030.0\n",
            "end             1.0\n",
            "Name: 30, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000031.0\n",
            "end             1.0\n",
            "Name: 31, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000032.0\n",
            "end             1.0\n",
            "Name: 32, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000033.0\n",
            "end             1.0\n",
            "Name: 33, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000034.0\n",
            "end             1.0\n",
            "Name: 34, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000035.0\n",
            "end             1.0\n",
            "Name: 35, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000036.0\n",
            "end             1.0\n",
            "Name: 36, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000037.0\n",
            "end             1.0\n",
            "Name: 37, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000038.0\n",
            "end             1.0\n",
            "Name: 38, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000039.0\n",
            "end             1.0\n",
            "Name: 39, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000040.0\n",
            "end             1.0\n",
            "Name: 40, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000041.0\n",
            "end             1.0\n",
            "Name: 41, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000042.0\n",
            "end             1.0\n",
            "Name: 42, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000043.0\n",
            "end             1.0\n",
            "Name: 43, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000044.0\n",
            "end             1.0\n",
            "Name: 44, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000045.0\n",
            "end             1.0\n",
            "Name: 45, dtype: float64\n",
            "chrom         NaN\n",
            "start    462057.0\n",
            "end      462073.0\n",
            "Name: 46, dtype: float64\n",
            "chrom         NaN\n",
            "start    481042.0\n",
            "end      481059.0\n",
            "Name: 47, dtype: float64\n",
            "chrom         NaN\n",
            "start    549918.0\n",
            "end      549929.0\n",
            "Name: 48, dtype: float64\n",
            "chrom         NaN\n",
            "start    549975.0\n",
            "end      549996.0\n",
            "Name: 49, dtype: float64\n",
            "chrom         NaN\n",
            "start    552457.0\n",
            "end      552480.0\n",
            "Name: 50, dtype: float64\n",
            "chrom         NaN\n",
            "start    552523.0\n",
            "end      552538.0\n",
            "Name: 51, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000046.0\n",
            "end             1.0\n",
            "Name: 52, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000047.0\n",
            "end             1.0\n",
            "Name: 53, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000048.0\n",
            "end             1.0\n",
            "Name: 54, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000049.0\n",
            "end             1.0\n",
            "Name: 55, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000050.0\n",
            "end             1.0\n",
            "Name: 56, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000051.0\n",
            "end             1.0\n",
            "Name: 57, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000052.0\n",
            "end             1.0\n",
            "Name: 58, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000053.0\n",
            "end             1.0\n",
            "Name: 59, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000054.0\n",
            "end             1.0\n",
            "Name: 60, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000055.0\n",
            "end             1.0\n",
            "Name: 61, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000056.0\n",
            "end             1.0\n",
            "Name: 62, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000057.0\n",
            "end             1.0\n",
            "Name: 63, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000058.0\n",
            "end             1.0\n",
            "Name: 64, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000059.0\n",
            "end             1.0\n",
            "Name: 65, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000060.0\n",
            "end             1.0\n",
            "Name: 66, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000061.0\n",
            "end             1.0\n",
            "Name: 67, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000062.0\n",
            "end             1.0\n",
            "Name: 68, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000063.0\n",
            "end             1.0\n",
            "Name: 69, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000064.0\n",
            "end             1.0\n",
            "Name: 70, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000065.0\n",
            "end             1.0\n",
            "Name: 71, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000066.0\n",
            "end             1.0\n",
            "Name: 72, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000067.0\n",
            "end             1.0\n",
            "Name: 73, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000068.0\n",
            "end             1.0\n",
            "Name: 74, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000069.0\n",
            "end             1.0\n",
            "Name: 75, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000070.0\n",
            "end             1.0\n",
            "Name: 76, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000071.0\n",
            "end             1.0\n",
            "Name: 77, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000072.0\n",
            "end             1.0\n",
            "Name: 78, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000073.0\n",
            "end             1.0\n",
            "Name: 79, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000074.0\n",
            "end             1.0\n",
            "Name: 80, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000075.0\n",
            "end             1.0\n",
            "Name: 81, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000076.0\n",
            "end             1.0\n",
            "Name: 82, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000077.0\n",
            "end             1.0\n",
            "Name: 83, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000078.0\n",
            "end             1.0\n",
            "Name: 84, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000079.0\n",
            "end             1.0\n",
            "Name: 85, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000080.0\n",
            "end             1.0\n",
            "Name: 86, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000081.0\n",
            "end             1.0\n",
            "Name: 87, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000082.0\n",
            "end             1.0\n",
            "Name: 88, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000083.0\n",
            "end             1.0\n",
            "Name: 89, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000084.0\n",
            "end             1.0\n",
            "Name: 90, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000085.0\n",
            "end             1.0\n",
            "Name: 91, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000086.0\n",
            "end             1.0\n",
            "Name: 92, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000087.0\n",
            "end             1.0\n",
            "Name: 93, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000088.0\n",
            "end             1.0\n",
            "Name: 94, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000089.0\n",
            "end             1.0\n",
            "Name: 95, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000090.0\n",
            "end             1.0\n",
            "Name: 96, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000091.0\n",
            "end             1.0\n",
            "Name: 97, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000092.0\n",
            "end             1.0\n",
            "Name: 98, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000093.0\n",
            "end             1.0\n",
            "Name: 99, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000094.0\n",
            "end             1.0\n",
            "Name: 100, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000095.0\n",
            "end             1.0\n",
            "Name: 101, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000096.0\n",
            "end             1.0\n",
            "Name: 102, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000097.0\n",
            "end             1.0\n",
            "Name: 103, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000098.0\n",
            "end             1.0\n",
            "Name: 104, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000099.0\n",
            "end             1.0\n",
            "Name: 105, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000100.0\n",
            "end             1.0\n",
            "Name: 106, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000101.0\n",
            "end             1.0\n",
            "Name: 107, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000102.0\n",
            "end             1.0\n",
            "Name: 108, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000103.0\n",
            "end             1.0\n",
            "Name: 109, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000104.0\n",
            "end             1.0\n",
            "Name: 110, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000105.0\n",
            "end             1.0\n",
            "Name: 111, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000106.0\n",
            "end             1.0\n",
            "Name: 112, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000107.0\n",
            "end             1.0\n",
            "Name: 113, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000108.0\n",
            "end             1.0\n",
            "Name: 114, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000109.0\n",
            "end             1.0\n",
            "Name: 115, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000110.0\n",
            "end             1.0\n",
            "Name: 116, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000111.0\n",
            "end             1.0\n",
            "Name: 117, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000112.0\n",
            "end             1.0\n",
            "Name: 118, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000113.0\n",
            "end             1.0\n",
            "Name: 119, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000114.0\n",
            "end             1.0\n",
            "Name: 120, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000115.0\n",
            "end             1.0\n",
            "Name: 121, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000116.0\n",
            "end             1.0\n",
            "Name: 122, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000117.0\n",
            "end             1.0\n",
            "Name: 123, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000118.0\n",
            "end             1.0\n",
            "Name: 124, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000119.0\n",
            "end             1.0\n",
            "Name: 125, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000120.0\n",
            "end             1.0\n",
            "Name: 126, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000121.0\n",
            "end             1.0\n",
            "Name: 127, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000122.0\n",
            "end             1.0\n",
            "Name: 128, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000123.0\n",
            "end             1.0\n",
            "Name: 129, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000124.0\n",
            "end             1.0\n",
            "Name: 130, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000125.0\n",
            "end             1.0\n",
            "Name: 131, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000126.0\n",
            "end             1.0\n",
            "Name: 132, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000127.0\n",
            "end             1.0\n",
            "Name: 133, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000128.0\n",
            "end             1.0\n",
            "Name: 134, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000129.0\n",
            "end             1.0\n",
            "Name: 135, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000130.0\n",
            "end             1.0\n",
            "Name: 136, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000131.0\n",
            "end             1.0\n",
            "Name: 137, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000132.0\n",
            "end             1.0\n",
            "Name: 138, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000133.0\n",
            "end             1.0\n",
            "Name: 139, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000134.0\n",
            "end             1.0\n",
            "Name: 140, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000135.0\n",
            "end             1.0\n",
            "Name: 141, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000136.0\n",
            "end             1.0\n",
            "Name: 142, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000137.0\n",
            "end             1.0\n",
            "Name: 143, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000138.0\n",
            "end             1.0\n",
            "Name: 144, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000139.0\n",
            "end             1.0\n",
            "Name: 145, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000140.0\n",
            "end             1.0\n",
            "Name: 146, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000141.0\n",
            "end             1.0\n",
            "Name: 147, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000142.0\n",
            "end             1.0\n",
            "Name: 148, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000143.0\n",
            "end             1.0\n",
            "Name: 149, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000144.0\n",
            "end             1.0\n",
            "Name: 150, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000145.0\n",
            "end             1.0\n",
            "Name: 151, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000146.0\n",
            "end             1.0\n",
            "Name: 152, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000147.0\n",
            "end             1.0\n",
            "Name: 153, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000148.0\n",
            "end             1.0\n",
            "Name: 154, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000149.0\n",
            "end             1.0\n",
            "Name: 155, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000150.0\n",
            "end             1.0\n",
            "Name: 156, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000151.0\n",
            "end             1.0\n",
            "Name: 157, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000152.0\n",
            "end             1.0\n",
            "Name: 158, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000153.0\n",
            "end             1.0\n",
            "Name: 159, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000154.0\n",
            "end             1.0\n",
            "Name: 160, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000155.0\n",
            "end             1.0\n",
            "Name: 161, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000156.0\n",
            "end             1.0\n",
            "Name: 162, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000157.0\n",
            "end             1.0\n",
            "Name: 163, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000158.0\n",
            "end             1.0\n",
            "Name: 164, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000159.0\n",
            "end             1.0\n",
            "Name: 165, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000160.0\n",
            "end             1.0\n",
            "Name: 166, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000161.0\n",
            "end             1.0\n",
            "Name: 167, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000162.0\n",
            "end             1.0\n",
            "Name: 168, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000163.0\n",
            "end             1.0\n",
            "Name: 169, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000164.0\n",
            "end             1.0\n",
            "Name: 170, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000165.0\n",
            "end             1.0\n",
            "Name: 171, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000166.0\n",
            "end             1.0\n",
            "Name: 172, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000167.0\n",
            "end             1.0\n",
            "Name: 173, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000168.0\n",
            "end             1.0\n",
            "Name: 174, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000169.0\n",
            "end             1.0\n",
            "Name: 175, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000170.0\n",
            "end             1.0\n",
            "Name: 176, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000171.0\n",
            "end             1.0\n",
            "Name: 177, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000172.0\n",
            "end             1.0\n",
            "Name: 178, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000173.0\n",
            "end             1.0\n",
            "Name: 179, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000174.0\n",
            "end             1.0\n",
            "Name: 180, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000175.0\n",
            "end             1.0\n",
            "Name: 181, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000176.0\n",
            "end             1.0\n",
            "Name: 182, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000177.0\n",
            "end             1.0\n",
            "Name: 183, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000178.0\n",
            "end             1.0\n",
            "Name: 184, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000179.0\n",
            "end             1.0\n",
            "Name: 185, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000180.0\n",
            "end             1.0\n",
            "Name: 186, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000181.0\n",
            "end             1.0\n",
            "Name: 187, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000182.0\n",
            "end             1.0\n",
            "Name: 188, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000183.0\n",
            "end             1.0\n",
            "Name: 189, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000184.0\n",
            "end             1.0\n",
            "Name: 190, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000185.0\n",
            "end             1.0\n",
            "Name: 191, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000186.0\n",
            "end             1.0\n",
            "Name: 192, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000187.0\n",
            "end             1.0\n",
            "Name: 193, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000188.0\n",
            "end             1.0\n",
            "Name: 194, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000189.0\n",
            "end             1.0\n",
            "Name: 195, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000190.0\n",
            "end             1.0\n",
            "Name: 196, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000191.0\n",
            "end             1.0\n",
            "Name: 197, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000192.0\n",
            "end             1.0\n",
            "Name: 198, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000193.0\n",
            "end             1.0\n",
            "Name: 199, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000194.0\n",
            "end             1.0\n",
            "Name: 200, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000195.0\n",
            "end             1.0\n",
            "Name: 201, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000196.0\n",
            "end             1.0\n",
            "Name: 202, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000197.0\n",
            "end             1.0\n",
            "Name: 203, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000198.0\n",
            "end             1.0\n",
            "Name: 204, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000199.0\n",
            "end             1.0\n",
            "Name: 205, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000200.0\n",
            "end             1.0\n",
            "Name: 206, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000201.0\n",
            "end             1.0\n",
            "Name: 207, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000202.0\n",
            "end             1.0\n",
            "Name: 208, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000203.0\n",
            "end             1.0\n",
            "Name: 209, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000204.0\n",
            "end             1.0\n",
            "Name: 210, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000205.0\n",
            "end             1.0\n",
            "Name: 211, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000206.0\n",
            "end             1.0\n",
            "Name: 212, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000207.0\n",
            "end             1.0\n",
            "Name: 213, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000208.0\n",
            "end             1.0\n",
            "Name: 214, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000209.0\n",
            "end             1.0\n",
            "Name: 215, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000210.0\n",
            "end             1.0\n",
            "Name: 216, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000211.0\n",
            "end             1.0\n",
            "Name: 217, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000212.0\n",
            "end             1.0\n",
            "Name: 218, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000213.0\n",
            "end             1.0\n",
            "Name: 219, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000214.0\n",
            "end             1.0\n",
            "Name: 220, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000215.0\n",
            "end             1.0\n",
            "Name: 221, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000216.0\n",
            "end             1.0\n",
            "Name: 222, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000217.0\n",
            "end             1.0\n",
            "Name: 223, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000218.0\n",
            "end             1.0\n",
            "Name: 224, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000219.0\n",
            "end             1.0\n",
            "Name: 225, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000220.0\n",
            "end             1.0\n",
            "Name: 226, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000221.0\n",
            "end             1.0\n",
            "Name: 227, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000222.0\n",
            "end             1.0\n",
            "Name: 228, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000223.0\n",
            "end             1.0\n",
            "Name: 229, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000224.0\n",
            "end             1.0\n",
            "Name: 230, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000225.0\n",
            "end             1.0\n",
            "Name: 231, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000226.0\n",
            "end             1.0\n",
            "Name: 232, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000227.0\n",
            "end             1.0\n",
            "Name: 233, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000228.0\n",
            "end             1.0\n",
            "Name: 234, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000229.0\n",
            "end             1.0\n",
            "Name: 235, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000230.0\n",
            "end             1.0\n",
            "Name: 236, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000231.0\n",
            "end             1.0\n",
            "Name: 237, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000232.0\n",
            "end             1.0\n",
            "Name: 238, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000233.0\n",
            "end             1.0\n",
            "Name: 239, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000234.0\n",
            "end             1.0\n",
            "Name: 240, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000235.0\n",
            "end             1.0\n",
            "Name: 241, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000236.0\n",
            "end             1.0\n",
            "Name: 242, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000237.0\n",
            "end             1.0\n",
            "Name: 243, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000238.0\n",
            "end             1.0\n",
            "Name: 244, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000239.0\n",
            "end             1.0\n",
            "Name: 245, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000240.0\n",
            "end             1.0\n",
            "Name: 246, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000241.0\n",
            "end             1.0\n",
            "Name: 247, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000242.0\n",
            "end             1.0\n",
            "Name: 248, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000243.0\n",
            "end             1.0\n",
            "Name: 249, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000244.0\n",
            "end             1.0\n",
            "Name: 250, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000245.0\n",
            "end             1.0\n",
            "Name: 251, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000246.0\n",
            "end             1.0\n",
            "Name: 252, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000247.0\n",
            "end             1.0\n",
            "Name: 253, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000248.0\n",
            "end             1.0\n",
            "Name: 254, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000249.0\n",
            "end             1.0\n",
            "Name: 255, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000250.0\n",
            "end             1.0\n",
            "Name: 256, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000251.0\n",
            "end             1.0\n",
            "Name: 257, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000252.0\n",
            "end             1.0\n",
            "Name: 258, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000253.0\n",
            "end             1.0\n",
            "Name: 259, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000254.0\n",
            "end             1.0\n",
            "Name: 260, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000255.0\n",
            "end             1.0\n",
            "Name: 261, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000256.0\n",
            "end             1.0\n",
            "Name: 262, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000257.0\n",
            "end             1.0\n",
            "Name: 263, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000258.0\n",
            "end             1.0\n",
            "Name: 264, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000259.0\n",
            "end             1.0\n",
            "Name: 265, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000260.0\n",
            "end             1.0\n",
            "Name: 266, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000261.0\n",
            "end             1.0\n",
            "Name: 267, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000262.0\n",
            "end             1.0\n",
            "Name: 268, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000263.0\n",
            "end             1.0\n",
            "Name: 269, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000264.0\n",
            "end             1.0\n",
            "Name: 270, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000265.0\n",
            "end             1.0\n",
            "Name: 271, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000266.0\n",
            "end             1.0\n",
            "Name: 272, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000267.0\n",
            "end             1.0\n",
            "Name: 273, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000268.0\n",
            "end             1.0\n",
            "Name: 274, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000269.0\n",
            "end             1.0\n",
            "Name: 275, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000270.0\n",
            "end             1.0\n",
            "Name: 276, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000271.0\n",
            "end             1.0\n",
            "Name: 277, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000272.0\n",
            "end             1.0\n",
            "Name: 278, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000273.0\n",
            "end             1.0\n",
            "Name: 279, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000274.0\n",
            "end             1.0\n",
            "Name: 280, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000275.0\n",
            "end             1.0\n",
            "Name: 281, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000276.0\n",
            "end             1.0\n",
            "Name: 282, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000277.0\n",
            "end             1.0\n",
            "Name: 283, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000278.0\n",
            "end             1.0\n",
            "Name: 284, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000279.0\n",
            "end             1.0\n",
            "Name: 285, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000280.0\n",
            "end             1.0\n",
            "Name: 286, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000281.0\n",
            "end             1.0\n",
            "Name: 287, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000282.0\n",
            "end             1.0\n",
            "Name: 288, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000283.0\n",
            "end             1.0\n",
            "Name: 289, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000284.0\n",
            "end             1.0\n",
            "Name: 290, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000285.0\n",
            "end             1.0\n",
            "Name: 291, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000286.0\n",
            "end             1.0\n",
            "Name: 292, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000287.0\n",
            "end             1.0\n",
            "Name: 293, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000288.0\n",
            "end             1.0\n",
            "Name: 294, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000289.0\n",
            "end             1.0\n",
            "Name: 295, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000290.0\n",
            "end             1.0\n",
            "Name: 296, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000291.0\n",
            "end             1.0\n",
            "Name: 297, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000292.0\n",
            "end             1.0\n",
            "Name: 298, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000293.0\n",
            "end             1.0\n",
            "Name: 299, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000294.0\n",
            "end             1.0\n",
            "Name: 300, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000295.0\n",
            "end             1.0\n",
            "Name: 301, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000296.0\n",
            "end             1.0\n",
            "Name: 302, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000297.0\n",
            "end             1.0\n",
            "Name: 303, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000298.0\n",
            "end             1.0\n",
            "Name: 304, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000299.0\n",
            "end             1.0\n",
            "Name: 305, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000300.0\n",
            "end             1.0\n",
            "Name: 306, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000301.0\n",
            "end             1.0\n",
            "Name: 307, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000302.0\n",
            "end             1.0\n",
            "Name: 308, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000303.0\n",
            "end             1.0\n",
            "Name: 309, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000304.0\n",
            "end             1.0\n",
            "Name: 310, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000305.0\n",
            "end             1.0\n",
            "Name: 311, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000306.0\n",
            "end             1.0\n",
            "Name: 312, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000307.0\n",
            "end             1.0\n",
            "Name: 313, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000308.0\n",
            "end             1.0\n",
            "Name: 314, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000309.0\n",
            "end             1.0\n",
            "Name: 315, dtype: float64\n",
            "chrom           NaN\n",
            "start    10000310.0\n",
            "end             1.0\n",
            "Name: 316, dtype: float64\n",
            "0 0 0 0 0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "zdna_counts, zdna_genes"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EXzjhfP0IvCK",
        "outputId": "38c48821-d33e-450e-f52f-84a438b0aa47"
      },
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "([0, 0, 0, 0, 0], {})"
            ]
          },
          "metadata": {},
          "execution_count": 32
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Тоже самое для квадруплексов"
      ],
      "metadata": {
        "id": "5MFI1xU3fTdv"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pqs_counts, pqs_genes = find_count(genome, pqs)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NStXnUNIJE0c",
        "outputId": "964b285f-233f-49c0-9dd4-c8e69b2d88f3"
      },
      "execution_count": 33,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chrom    CAJJDO010000002.1\n",
            "start               534196\n",
            "end                 534314\n",
            "Name: 0, dtype: object\n",
            "chrom    CAJJDO010000009.1\n",
            "start               519367\n",
            "end                 519397\n",
            "Name: 1, dtype: object\n",
            "chrom    CAJJDO010000010.1\n",
            "start                67153\n",
            "end                  67174\n",
            "Name: 2, dtype: object\n",
            "chrom    CAJJDO010000012.1\n",
            "start               289822\n",
            "end                 289851\n",
            "Name: 3, dtype: object\n",
            "chrom    CAJJDO010000012.1\n",
            "start                    0\n",
            "end                     76\n",
            "Name: 4, dtype: object\n",
            "chrom    CAJJDO010000016.1\n",
            "start               354394\n",
            "end                 354442\n",
            "Name: 5, dtype: object\n",
            "chrom    CAJJDO010000020.1\n",
            "start               327022\n",
            "end                 327079\n",
            "Name: 6, dtype: object\n",
            "chrom    CAJJDO010000021.1\n",
            "start               754529\n",
            "end                 754592\n",
            "Name: 7, dtype: object\n",
            "chrom    CAJJDO010000021.1\n",
            "start               755057\n",
            "end                 755083\n",
            "Name: 8, dtype: object\n",
            "chrom    CAJJDO010000022.1\n",
            "start                34858\n",
            "end                  34945\n",
            "Name: 9, dtype: object\n",
            "chrom    CAJJDO010000023.1\n",
            "start               162532\n",
            "end                 162562\n",
            "Name: 10, dtype: object\n",
            "chrom    CAJJDO010000024.1\n",
            "start                 1084\n",
            "end                   1111\n",
            "Name: 11, dtype: object\n",
            "chrom    CAJJDO010000024.1\n",
            "start               612113\n",
            "end                 612134\n",
            "Name: 12, dtype: object\n",
            "chrom    CAJJDO010000026.1\n",
            "start               337492\n",
            "end                 337515\n",
            "Name: 13, dtype: object\n",
            "chrom    CAJJDO010000026.1\n",
            "start                    0\n",
            "end                     21\n",
            "Name: 14, dtype: object\n",
            "chrom    CAJJDO010000027.1\n",
            "start                49282\n",
            "end                  49377\n",
            "Name: 15, dtype: object\n",
            "chrom    CAJJDO010000028.1\n",
            "start               552192\n",
            "end                 552249\n",
            "Name: 16, dtype: object\n",
            "chrom    CAJJDO010000039.1\n",
            "start               232436\n",
            "end                 232457\n",
            "Name: 17, dtype: object\n",
            "chrom    CAJJDO010000040.1\n",
            "start               592964\n",
            "end                 593033\n",
            "Name: 18, dtype: object\n",
            "chrom    CAJJDO010000042.1\n",
            "start                11000\n",
            "end                  11069\n",
            "Name: 19, dtype: object\n",
            "chrom    CAJJDO010000045.1\n",
            "start               456610\n",
            "end                 456644\n",
            "Name: 20, dtype: object\n",
            "chrom    CAJJDO010000045.1\n",
            "start               501770\n",
            "end                 501785\n",
            "Name: 21, dtype: object\n",
            "chrom    CAJJDO010000048.1\n",
            "start                46848\n",
            "end                  46929\n",
            "Name: 22, dtype: object\n",
            "chrom    CAJJDO010000048.1\n",
            "start               102321\n",
            "end                 102343\n",
            "Name: 23, dtype: object\n",
            "chrom    CAJJDO010000053.1\n",
            "start                    0\n",
            "end                     69\n",
            "Name: 24, dtype: object\n",
            "chrom    CAJJDO010000054.1\n",
            "start               348279\n",
            "end                 348310\n",
            "Name: 25, dtype: object\n",
            "chrom    CAJJDO010000061.1\n",
            "start               496520\n",
            "end                 496625\n",
            "Name: 26, dtype: object\n",
            "chrom    CAJJDO010000062.1\n",
            "start               385085\n",
            "end                 385112\n",
            "Name: 27, dtype: object\n",
            "chrom    CAJJDO010000062.1\n",
            "start               515135\n",
            "end                 515159\n",
            "Name: 28, dtype: object\n",
            "chrom    CAJJDO010000063.1\n",
            "start                19246\n",
            "end                  19315\n",
            "Name: 29, dtype: object\n",
            "chrom    CAJJDO010000066.1\n",
            "start               379145\n",
            "end                 379275\n",
            "Name: 30, dtype: object\n",
            "chrom    CAJJDO010000068.1\n",
            "start               496520\n",
            "end                 496541\n",
            "Name: 31, dtype: object\n",
            "chrom    CAJJDO010000068.1\n",
            "start                    0\n",
            "end                     45\n",
            "Name: 32, dtype: object\n",
            "chrom    CAJJDO010000068.1\n",
            "start                  794\n",
            "end                    819\n",
            "Name: 33, dtype: object\n",
            "chrom    CAJJDO010000071.1\n",
            "start               300485\n",
            "end                 300608\n",
            "Name: 34, dtype: object\n",
            "chrom    CAJJDO010000071.1\n",
            "start               307157\n",
            "end                 307238\n",
            "Name: 35, dtype: object\n",
            "chrom    CAJJDO010000072.1\n",
            "start               232078\n",
            "end                 232153\n",
            "Name: 36, dtype: object\n",
            "chrom    CAJJDO010000073.1\n",
            "start               431762\n",
            "end                 431898\n",
            "Name: 37, dtype: object\n",
            "chrom    CAJJDO010000076.1\n",
            "start                 5109\n",
            "end                   5179\n",
            "Name: 38, dtype: object\n",
            "chrom    CAJJDO010000077.1\n",
            "start                    3\n",
            "end                     50\n",
            "Name: 39, dtype: object\n",
            "chrom    CAJJDO010000082.1\n",
            "start               458705\n",
            "end                 458738\n",
            "Name: 40, dtype: object\n",
            "chrom    CAJJDO010000085.1\n",
            "start               444911\n",
            "end                 444938\n",
            "Name: 41, dtype: object\n",
            "chrom    CAJJDO010000087.1\n",
            "start               384079\n",
            "end                 384096\n",
            "Name: 42, dtype: object\n",
            "chrom    CAJJDO010000093.1\n",
            "start               351441\n",
            "end                 351480\n",
            "Name: 43, dtype: object\n",
            "chrom    CAJJDO010000098.1\n",
            "start                    4\n",
            "end                     43\n",
            "Name: 44, dtype: object\n",
            "chrom    CAJJDO010000098.1\n",
            "start               380860\n",
            "end                 380923\n",
            "Name: 45, dtype: object\n",
            "chrom    CAJJDO010000099.1\n",
            "start               386675\n",
            "end                 386757\n",
            "Name: 46, dtype: object\n",
            "chrom    CAJJDO010000106.1\n",
            "start               393264\n",
            "end                 393315\n",
            "Name: 47, dtype: object\n",
            "chrom    CAJJDO010000107.1\n",
            "start                30513\n",
            "end                  30606\n",
            "Name: 48, dtype: object\n",
            "chrom    CAJJDO010000110.1\n",
            "start                26848\n",
            "end                  26903\n",
            "Name: 49, dtype: object\n",
            "chrom    CAJJDO010000110.1\n",
            "start                26995\n",
            "end                  27140\n",
            "Name: 50, dtype: object\n",
            "chrom    CAJJDO010000112.1\n",
            "start                57843\n",
            "end                  57876\n",
            "Name: 51, dtype: object\n",
            "chrom    CAJJDO010000114.1\n",
            "start               365582\n",
            "end                 365657\n",
            "Name: 52, dtype: object\n",
            "chrom    CAJJDO010000114.1\n",
            "start                    0\n",
            "end                     21\n",
            "Name: 53, dtype: object\n",
            "chrom    CAJJDO010000120.1\n",
            "start                12792\n",
            "end                  12875\n",
            "Name: 54, dtype: object\n",
            "chrom    CAJJDO010000120.1\n",
            "start                53230\n",
            "end                  53251\n",
            "Name: 55, dtype: object\n",
            "chrom    CAJJDO010000120.1\n",
            "start                54430\n",
            "end                  54451\n",
            "Name: 56, dtype: object\n",
            "chrom    CAJJDO010000124.1\n",
            "start               342610\n",
            "end                 342638\n",
            "Name: 57, dtype: object\n",
            "chrom    CAJJDO010000124.1\n",
            "start                    0\n",
            "end                    113\n",
            "Name: 58, dtype: object\n",
            "chrom    CAJJDO010000126.1\n",
            "start               334992\n",
            "end                 335158\n",
            "Name: 59, dtype: object\n",
            "chrom    CAJJDO010000126.1\n",
            "start               341480\n",
            "end                 341549\n",
            "Name: 60, dtype: object\n",
            "chrom    CAJJDO010000127.1\n",
            "start               271802\n",
            "end                 271919\n",
            "Name: 61, dtype: object\n",
            "chrom    CAJJDO010000130.1\n",
            "start               143366\n",
            "end                 143405\n",
            "Name: 62, dtype: object\n",
            "chrom    CAJJDO010000130.1\n",
            "start               180005\n",
            "end                 180034\n",
            "Name: 63, dtype: object\n",
            "chrom    CAJJDO010000135.1\n",
            "start               291502\n",
            "end                 291565\n",
            "Name: 64, dtype: object\n",
            "chrom    CAJJDO010000136.1\n",
            "start               155144\n",
            "end                 155177\n",
            "Name: 65, dtype: object\n",
            "chrom    CAJJDO010000136.1\n",
            "start               184590\n",
            "end                 184622\n",
            "Name: 66, dtype: object\n",
            "chrom    CAJJDO010000137.1\n",
            "start                91604\n",
            "end                  91655\n",
            "Name: 67, dtype: object\n",
            "chrom    CAJJDO010000139.1\n",
            "start               316225\n",
            "end                 316242\n",
            "Name: 68, dtype: object\n",
            "chrom    CAJJDO010000141.1\n",
            "start               280656\n",
            "end                 280677\n",
            "Name: 69, dtype: object\n",
            "chrom    CAJJDO010000148.1\n",
            "start                 8404\n",
            "end                   8521\n",
            "Name: 70, dtype: object\n",
            "chrom    CAJJDO010000148.1\n",
            "start               283756\n",
            "end                 283783\n",
            "Name: 71, dtype: object\n",
            "chrom    CAJJDO010000148.1\n",
            "start                 8965\n",
            "end                   8992\n",
            "Name: 72, dtype: object\n",
            "chrom    CAJJDO010000149.1\n",
            "start                    1\n",
            "end                     59\n",
            "Name: 73, dtype: object\n",
            "chrom    CAJJDO010000163.1\n",
            "start               192077\n",
            "end                 192104\n",
            "Name: 74, dtype: object\n",
            "chrom    CAJJDO010000163.1\n",
            "start               192602\n",
            "end                 192629\n",
            "Name: 75, dtype: object\n",
            "chrom    CAJJDO010000169.1\n",
            "start                11130\n",
            "end                  11163\n",
            "Name: 76, dtype: object\n",
            "chrom    CAJJDO010000169.1\n",
            "start               110999\n",
            "end                 111032\n",
            "Name: 77, dtype: object\n",
            "chrom    CAJJDO010000172.1\n",
            "start                    0\n",
            "end                     22\n",
            "Name: 78, dtype: object\n",
            "chrom    CAJJDO010000182.1\n",
            "start                 1216\n",
            "end                   1255\n",
            "Name: 79, dtype: object\n",
            "chrom    CAJJDO010000183.1\n",
            "start                47996\n",
            "end                  48059\n",
            "Name: 80, dtype: object\n",
            "chrom    CAJJDO010000184.1\n",
            "start                 8714\n",
            "end                   8747\n",
            "Name: 81, dtype: object\n",
            "chrom    CAJJDO010000187.1\n",
            "start                63620\n",
            "end                  63666\n",
            "Name: 82, dtype: object\n",
            "chrom    CAJJDO010000192.1\n",
            "start                47494\n",
            "end                  47570\n",
            "Name: 83, dtype: object\n",
            "chrom    CAJJDO010000197.1\n",
            "start                18783\n",
            "end                  18804\n",
            "Name: 84, dtype: object\n",
            "chrom    CAJJDO010000197.1\n",
            "start                 4949\n",
            "end                   4988\n",
            "Name: 85, dtype: object\n",
            "chrom    CAJJDO010000222.1\n",
            "start                 3603\n",
            "end                   3643\n",
            "Name: 86, dtype: object\n",
            "chrom    CAJJDO010000229.1\n",
            "start                    0\n",
            "end                     21\n",
            "Name: 87, dtype: object\n",
            "chrom    CAJJDO010000242.1\n",
            "start                 8289\n",
            "end                   8376\n",
            "Name: 88, dtype: object\n",
            "chrom    CAJJDO010000254.1\n",
            "start                10069\n",
            "end                  10096\n",
            "Name: 89, dtype: object\n",
            "chrom    CAJJDO010000259.1\n",
            "start                 7681\n",
            "end                   7757\n",
            "Name: 90, dtype: object\n",
            "0 0 0 0 0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "pqs_counts, pqs_genes"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "oynGuQlOKLc2",
        "outputId": "e7a553c7-ef89-4425-d507-22a34db06287"
      },
      "execution_count": 34,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "([0, 0, 0, 0, 0], {})"
            ]
          },
          "metadata": {},
          "execution_count": 34
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Zhunt сделал в другом блокноте"
      ],
      "metadata": {
        "id": "SdRB_3OPfiWl"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Теперь получим наш вывод hmmer и переведём его в pandas"
      ],
      "metadata": {
        "id": "Qu7_jR-pgOUX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "from collections import defaultdict\n",
        "from Bio import SearchIO\n",
        "\n",
        "filename = 'filtered.out'\n",
        "\n",
        "attribs = ['accession', 'bias', 'bitscore', 'description', 'cluster_num', 'domain_exp_num',  'domain_included_num', 'domain_obs_num', 'domain_reported_num', 'env_num', 'evalue', 'id', 'query_id', 'overlap_num', 'region_num']\n",
        "\n",
        "hits = defaultdict(list)\n",
        "\n",
        "with open(filename) as handle:\n",
        "    for queryresult in SearchIO.parse(handle, 'hmmer3-tab'):\n",
        "      for hit in queryresult.hits:\n",
        "        for attrib in attribs:\n",
        "          hits[attrib].append(getattr(hit, attrib))\n",
        "\n",
        "hmmer_data = pd.DataFrame.from_dict(hits)\n",
        "\n",
        "hmmer_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "RI6wgdPkVQIj",
        "outputId": "c5ff550d-8efd-4357-8c63-d260bdfb2d3a"
      },
      "execution_count": 37,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   accession   bias  bitscore  \\\n",
              "0          -  620.5    -330.2   \n",
              "1          -  298.3    -261.6   \n",
              "2          -  295.2    -189.2   \n",
              "3          -  416.7    -308.2   \n",
              "4          -  245.6    -224.3   \n",
              "5          -  340.9    -161.5   \n",
              "6          -  336.1    -252.8   \n",
              "7          -  265.9    -192.5   \n",
              "8          -  168.6    -152.0   \n",
              "9          -  380.4    -217.2   \n",
              "10         -  446.0    -187.4   \n",
              "11         -  122.7     -97.7   \n",
              "12         -  290.8    -195.6   \n",
              "13         -  157.1    -144.0   \n",
              "14         -  463.2    -259.3   \n",
              "15         -  158.8    -133.8   \n",
              "16         -  232.9    -144.1   \n",
              "17         -  557.0    -373.5   \n",
              "18         -  395.9    -301.5   \n",
              "19         -  285.6    -230.9   \n",
              "20         -  260.4    -183.5   \n",
              "21         -  270.5    -192.7   \n",
              "22         -  275.0    -212.5   \n",
              "23         -  278.1    -181.0   \n",
              "24         -  295.2    -213.1   \n",
              "25         -  345.8    -228.9   \n",
              "26         -  470.0    -383.2   \n",
              "27         -  291.2    -214.4   \n",
              "\n",
              "                                          description  cluster_num  \\\n",
              "0   Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "1   Paramecium pentaurelia isolate 4_5 large subun...            2   \n",
              "2   Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "3   Paramecium pentaurelia isolate 4_5 large subun...            4   \n",
              "4   Paramecium pentaurelia isolate 4_5 large subun...            4   \n",
              "5   Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "6   Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "7   Paramecium pentaurelia isolate 4_5 large subun...            2   \n",
              "8   Paramecium pentaurelia isolate 4_5 large subun...            0   \n",
              "9   Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "10  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "11  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "12  Paramecium pentaurelia isolate 4_5 large subun...            2   \n",
              "13  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "14  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "15  Paramecium pentaurelia isolate 4_5 large subun...            0   \n",
              "16  Paramecium pentaurelia isolate 4_5 large subun...            2   \n",
              "17  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "18  Paramecium pentaurelia isolate 4_5 large subun...            5   \n",
              "19  Paramecium pentaurelia isolate 4_5 large subun...            4   \n",
              "20  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "21  Paramecium pentaurelia isolate 4_5 large subun...            3   \n",
              "22  Paramecium pentaurelia isolate 4_5 large subun...            2   \n",
              "23  Paramecium pentaurelia isolate 4_5 large subun...            2   \n",
              "24  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "25  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "26  Paramecium pentaurelia isolate 4_5 large subun...           10   \n",
              "27  Paramecium pentaurelia isolate 4_5 large subun...            1   \n",
              "\n",
              "    domain_exp_num  domain_included_num  domain_obs_num  domain_reported_num  \\\n",
              "0             21.9                    0               9                    9   \n",
              "1              8.7                    0              11                   11   \n",
              "2             12.1                    0               3                    3   \n",
              "3             13.5                    0              13                   13   \n",
              "4              7.6                    0              11                   11   \n",
              "5              7.0                    0               6                    6   \n",
              "6             11.4                    0              12                   12   \n",
              "7             10.3                    0              11                   11   \n",
              "8              4.5                    0               6                    6   \n",
              "9             13.7                    0              12                   12   \n",
              "10            12.7                    0               8                    8   \n",
              "11             4.8                    0               6                    6   \n",
              "12             9.9                    0               9                    9   \n",
              "13             5.0                    0               3                    3   \n",
              "14             8.3                    0               7                    7   \n",
              "15             5.9                    0               6                    6   \n",
              "16            13.8                    0               5                    5   \n",
              "17            17.1                    0              12                   12   \n",
              "18            20.6                    0              10                   10   \n",
              "19            15.3                    0              13                   13   \n",
              "20             6.3                    0               6                    6   \n",
              "21            10.2                    0               8                    8   \n",
              "22            11.3                    0               9                    9   \n",
              "23            11.9                    0               6                    6   \n",
              "24            13.9                    0               5                    5   \n",
              "25            11.3                    0              10                   10   \n",
              "26            26.2                    0              25                   25   \n",
              "27            15.3                    0               3                    3   \n",
              "\n",
              "    env_num  evalue          id         query_id  overlap_num  region_num  \n",
              "0         9     1.0  KF287687.1  Bacteriocin_IIc            8           1  \n",
              "1        11     1.0  KF287687.1            BRINP            7           2  \n",
              "2         3     1.0  KF287687.1         Dehydrin            1           1  \n",
              "3        13     1.0  KF287687.1          DUF3482            8           5  \n",
              "4        11     1.0  KF287687.1          DUF4616            3           6  \n",
              "5         6     1.0  KF287687.1          DUF5585            4           1  \n",
              "6        12     1.0  KF287687.1           DUF726           11           1  \n",
              "7        11     1.0  KF287687.1            FSA_C            8           2  \n",
              "8         6     1.0  KF287687.1        GAGA_bind            0           6  \n",
              "9        12     1.0  KF287687.1    Herpes_capsid           11           1  \n",
              "10        8     1.0  KF287687.1   Keratin_2_head            7           1  \n",
              "11        6     1.0  KF287687.1            Med15            4           2  \n",
              "12        9     1.0  KF287687.1             Med3            7           2  \n",
              "13        3     1.0  KF287687.1         NMDAR2_C            0           3  \n",
              "14        7     1.0  KF287687.1  Nucleoporin_FG2            6           1  \n",
              "15        6     1.0  KF287687.1     Orthopox_A5L            0           6  \n",
              "16        5     1.0  KF287687.1          PPE-SVP            2           3  \n",
              "17       12     1.0  KF287687.1            PRR18           11           1  \n",
              "18       10     1.0  KF287687.1          PT-VENN            5           5  \n",
              "19       13     1.0  KF287687.1  Ribosomal_L12_N            0          13  \n",
              "20        6     1.0  KF287687.1   Sporozoite_P67            5           1  \n",
              "21        8     1.0  KF287687.1             SseC            4           4  \n",
              "22        9     1.0  KF287687.1           Tic110            6           2  \n",
              "23        6     1.0  KF287687.1          TPPII_N            4           2  \n",
              "24        5     1.0  KF287687.1             TraT            2           1  \n",
              "25       10     1.0  KF287687.1           TrbL_3            9           1  \n",
              "26       25     1.0  KF287687.1       UCR_Fe-S_N            5          20  \n",
              "27        3     1.0  KF287687.1             YqfQ            1           1  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-3da63d66-ea1d-4b17-a38f-cfb4ce50baf0\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>accession</th>\n",
              "      <th>bias</th>\n",
              "      <th>bitscore</th>\n",
              "      <th>description</th>\n",
              "      <th>cluster_num</th>\n",
              "      <th>domain_exp_num</th>\n",
              "      <th>domain_included_num</th>\n",
              "      <th>domain_obs_num</th>\n",
              "      <th>domain_reported_num</th>\n",
              "      <th>env_num</th>\n",
              "      <th>evalue</th>\n",
              "      <th>id</th>\n",
              "      <th>query_id</th>\n",
              "      <th>overlap_num</th>\n",
              "      <th>region_num</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>-</td>\n",
              "      <td>620.5</td>\n",
              "      <td>-330.2</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>21.9</td>\n",
              "      <td>0</td>\n",
              "      <td>9</td>\n",
              "      <td>9</td>\n",
              "      <td>9</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Bacteriocin_IIc</td>\n",
              "      <td>8</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>-</td>\n",
              "      <td>298.3</td>\n",
              "      <td>-261.6</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>2</td>\n",
              "      <td>8.7</td>\n",
              "      <td>0</td>\n",
              "      <td>11</td>\n",
              "      <td>11</td>\n",
              "      <td>11</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>BRINP</td>\n",
              "      <td>7</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>-</td>\n",
              "      <td>295.2</td>\n",
              "      <td>-189.2</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>12.1</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Dehydrin</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>-</td>\n",
              "      <td>416.7</td>\n",
              "      <td>-308.2</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>4</td>\n",
              "      <td>13.5</td>\n",
              "      <td>0</td>\n",
              "      <td>13</td>\n",
              "      <td>13</td>\n",
              "      <td>13</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>DUF3482</td>\n",
              "      <td>8</td>\n",
              "      <td>5</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>-</td>\n",
              "      <td>245.6</td>\n",
              "      <td>-224.3</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>4</td>\n",
              "      <td>7.6</td>\n",
              "      <td>0</td>\n",
              "      <td>11</td>\n",
              "      <td>11</td>\n",
              "      <td>11</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>DUF4616</td>\n",
              "      <td>3</td>\n",
              "      <td>6</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>-</td>\n",
              "      <td>340.9</td>\n",
              "      <td>-161.5</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>7.0</td>\n",
              "      <td>0</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>DUF5585</td>\n",
              "      <td>4</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>-</td>\n",
              "      <td>336.1</td>\n",
              "      <td>-252.8</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>11.4</td>\n",
              "      <td>0</td>\n",
              "      <td>12</td>\n",
              "      <td>12</td>\n",
              "      <td>12</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>DUF726</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>-</td>\n",
              "      <td>265.9</td>\n",
              "      <td>-192.5</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>2</td>\n",
              "      <td>10.3</td>\n",
              "      <td>0</td>\n",
              "      <td>11</td>\n",
              "      <td>11</td>\n",
              "      <td>11</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>FSA_C</td>\n",
              "      <td>8</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>-</td>\n",
              "      <td>168.6</td>\n",
              "      <td>-152.0</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>0</td>\n",
              "      <td>4.5</td>\n",
              "      <td>0</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>GAGA_bind</td>\n",
              "      <td>0</td>\n",
              "      <td>6</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>-</td>\n",
              "      <td>380.4</td>\n",
              "      <td>-217.2</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>13.7</td>\n",
              "      <td>0</td>\n",
              "      <td>12</td>\n",
              "      <td>12</td>\n",
              "      <td>12</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Herpes_capsid</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>-</td>\n",
              "      <td>446.0</td>\n",
              "      <td>-187.4</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>12.7</td>\n",
              "      <td>0</td>\n",
              "      <td>8</td>\n",
              "      <td>8</td>\n",
              "      <td>8</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Keratin_2_head</td>\n",
              "      <td>7</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>-</td>\n",
              "      <td>122.7</td>\n",
              "      <td>-97.7</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>4.8</td>\n",
              "      <td>0</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Med15</td>\n",
              "      <td>4</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>-</td>\n",
              "      <td>290.8</td>\n",
              "      <td>-195.6</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>2</td>\n",
              "      <td>9.9</td>\n",
              "      <td>0</td>\n",
              "      <td>9</td>\n",
              "      <td>9</td>\n",
              "      <td>9</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Med3</td>\n",
              "      <td>7</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>-</td>\n",
              "      <td>157.1</td>\n",
              "      <td>-144.0</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>5.0</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>NMDAR2_C</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>-</td>\n",
              "      <td>463.2</td>\n",
              "      <td>-259.3</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>8.3</td>\n",
              "      <td>0</td>\n",
              "      <td>7</td>\n",
              "      <td>7</td>\n",
              "      <td>7</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Nucleoporin_FG2</td>\n",
              "      <td>6</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>-</td>\n",
              "      <td>158.8</td>\n",
              "      <td>-133.8</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>0</td>\n",
              "      <td>5.9</td>\n",
              "      <td>0</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Orthopox_A5L</td>\n",
              "      <td>0</td>\n",
              "      <td>6</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>-</td>\n",
              "      <td>232.9</td>\n",
              "      <td>-144.1</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>2</td>\n",
              "      <td>13.8</td>\n",
              "      <td>0</td>\n",
              "      <td>5</td>\n",
              "      <td>5</td>\n",
              "      <td>5</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>PPE-SVP</td>\n",
              "      <td>2</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>-</td>\n",
              "      <td>557.0</td>\n",
              "      <td>-373.5</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>17.1</td>\n",
              "      <td>0</td>\n",
              "      <td>12</td>\n",
              "      <td>12</td>\n",
              "      <td>12</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>PRR18</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>-</td>\n",
              "      <td>395.9</td>\n",
              "      <td>-301.5</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>5</td>\n",
              "      <td>20.6</td>\n",
              "      <td>0</td>\n",
              "      <td>10</td>\n",
              "      <td>10</td>\n",
              "      <td>10</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>PT-VENN</td>\n",
              "      <td>5</td>\n",
              "      <td>5</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>-</td>\n",
              "      <td>285.6</td>\n",
              "      <td>-230.9</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>4</td>\n",
              "      <td>15.3</td>\n",
              "      <td>0</td>\n",
              "      <td>13</td>\n",
              "      <td>13</td>\n",
              "      <td>13</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Ribosomal_L12_N</td>\n",
              "      <td>0</td>\n",
              "      <td>13</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>-</td>\n",
              "      <td>260.4</td>\n",
              "      <td>-183.5</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>6.3</td>\n",
              "      <td>0</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Sporozoite_P67</td>\n",
              "      <td>5</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>-</td>\n",
              "      <td>270.5</td>\n",
              "      <td>-192.7</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>3</td>\n",
              "      <td>10.2</td>\n",
              "      <td>0</td>\n",
              "      <td>8</td>\n",
              "      <td>8</td>\n",
              "      <td>8</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>SseC</td>\n",
              "      <td>4</td>\n",
              "      <td>4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>-</td>\n",
              "      <td>275.0</td>\n",
              "      <td>-212.5</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>2</td>\n",
              "      <td>11.3</td>\n",
              "      <td>0</td>\n",
              "      <td>9</td>\n",
              "      <td>9</td>\n",
              "      <td>9</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>Tic110</td>\n",
              "      <td>6</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>-</td>\n",
              "      <td>278.1</td>\n",
              "      <td>-181.0</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>2</td>\n",
              "      <td>11.9</td>\n",
              "      <td>0</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>6</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>TPPII_N</td>\n",
              "      <td>4</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>-</td>\n",
              "      <td>295.2</td>\n",
              "      <td>-213.1</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>13.9</td>\n",
              "      <td>0</td>\n",
              "      <td>5</td>\n",
              "      <td>5</td>\n",
              "      <td>5</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>TraT</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>-</td>\n",
              "      <td>345.8</td>\n",
              "      <td>-228.9</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>11.3</td>\n",
              "      <td>0</td>\n",
              "      <td>10</td>\n",
              "      <td>10</td>\n",
              "      <td>10</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>TrbL_3</td>\n",
              "      <td>9</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>-</td>\n",
              "      <td>470.0</td>\n",
              "      <td>-383.2</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>10</td>\n",
              "      <td>26.2</td>\n",
              "      <td>0</td>\n",
              "      <td>25</td>\n",
              "      <td>25</td>\n",
              "      <td>25</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>UCR_Fe-S_N</td>\n",
              "      <td>5</td>\n",
              "      <td>20</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>-</td>\n",
              "      <td>291.2</td>\n",
              "      <td>-214.4</td>\n",
              "      <td>Paramecium pentaurelia isolate 4_5 large subun...</td>\n",
              "      <td>1</td>\n",
              "      <td>15.3</td>\n",
              "      <td>0</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>3</td>\n",
              "      <td>1.0</td>\n",
              "      <td>KF287687.1</td>\n",
              "      <td>YqfQ</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-3da63d66-ea1d-4b17-a38f-cfb4ce50baf0')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-3da63d66-ea1d-4b17-a38f-cfb4ce50baf0 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-3da63d66-ea1d-4b17-a38f-cfb4ce50baf0');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-375211cd-dc29-4d64-9676-74331b25d9a5\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-375211cd-dc29-4d64-9676-74331b25d9a5')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-375211cd-dc29-4d64-9676-74331b25d9a5 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "hmmer_data",
              "summary": "{\n  \"name\": \"hmmer_data\",\n  \"rows\": 28,\n  \"fields\": [\n    {\n      \"column\": \"accession\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"-\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bias\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 117.10100555778973,\n        \"min\": 122.7,\n        \"max\": 620.5,\n        \"num_unique_values\": 27,\n        \"samples\": [\n          168.6\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bitscore\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 69.49640659128016,\n        \"min\": -383.2,\n        \"max\": -97.7,\n        \"num_unique_values\": 28,\n        \"samples\": [\n          -217.2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"Paramecium pentaurelia isolate 4_5 large subunit ribosomal RNA gene, partial sequence\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"cluster_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2,\n        \"min\": 0,\n        \"max\": 10,\n        \"num_unique_values\": 7,\n        \"samples\": [\n          1\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_exp_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 5.236727996165173,\n        \"min\": 4.5,\n        \"max\": 26.2,\n        \"num_unique_values\": 26,\n        \"samples\": [\n          4.5\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_included_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0,\n        \"min\": 0,\n        \"max\": 0,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_obs_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 4,\n        \"min\": 3,\n        \"max\": 25,\n        \"num_unique_values\": 11,\n        \"samples\": [\n          12\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"domain_reported_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 4,\n        \"min\": 3,\n        \"max\": 25,\n        \"num_unique_values\": 11,\n        \"samples\": [\n          12\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"env_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 4,\n        \"min\": 3,\n        \"max\": 25,\n        \"num_unique_values\": 11,\n        \"samples\": [\n          12\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"evalue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.0,\n        \"min\": 1.0,\n        \"max\": 1.0,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          1.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"KF287687.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"query_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 28,\n        \"samples\": [\n          \"Herpes_capsid\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"overlap_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 3,\n        \"min\": 0,\n        \"max\": 11,\n        \"num_unique_values\": 11,\n        \"samples\": [\n          11\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"region_num\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 4,\n        \"min\": 1,\n        \"max\": 20,\n        \"num_unique_values\": 8,\n        \"samples\": [\n          2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 37
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "T4"
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.9.13"
    },
    "accelerator": "GPU"
  },
  "nbformat": 4,
  "nbformat_minor": 0
}